---
title: "RNAseq TCGA-SARC"
author: "Alicia Pliego"
output: 
  html_document:
    toc: true
    toc_float: true
    toc_collapsed: true
    toc_depth: 3
    number_sections: true
    theme: sandstone
---
<style>
body {
text-align: justify}
</style>

https://www.nature.com/articles/s41588-019-0576-7
https://github.com/parklab/ShatterSeek

```{r}
library(ShatterSeek)
data(DO17373)
```

```{r}
SV_data <- SVs(chrom1=as.character(SV_DO17373$chrom1), 
			   pos1=as.numeric(SV_DO17373$start1),
			   chrom2=as.character(SV_DO17373$chrom2), 
			   pos2=as.numeric(SV_DO17373$end2),
			   SVtype=as.character(SV_DO17373$svclass), 
			   strand1=as.character(SV_DO17373$strand1),
			   strand2=as.character(SV_DO17373$strand2))
```

```{r}
CN_data <- CNVsegs(chrom=as.character(SCNA_DO17373$chromosome),
				   start=SCNA_DO17373$start,
				   end=SCNA_DO17373$end,
				   total_cn=SCNA_DO17373$total_cn)
```

```{r}
chromothripsis <- shatterseek(
                SV.sample=SV_data,
                seg.sample=CN_data,
                genome="hg19")

```

```{r}
library(gridExtra)
library(cowplot)

plots_chr21 <- plot_chromothripsis(ShatterSeek_output = chromothripsis, 
              chr = "chr21", sample_name="DO17373", genome="hg19")
              
plot_chr21 = arrangeGrob(plots_chr21[[1]],
                         plots_chr21[[2]],
                         plots_chr21[[3]],
                         plots_chr21[[4]],
                         nrow=4,ncol=1,heights=c(0.2,.4,.4,.4))

plots_chrX <- plot_chromothripsis(ShatterSeek_output = chromothripsis, 
              chr = "chrX", sample_name="DO17373", genome="hg19")
plot_chrX = arrangeGrob(plots_chrX[[1]],
                        plots_chrX[[2]],
                        plots_chrX[[3]],
                        plots_chrX[[4]],
                         nrow=4,ncol=1,heights=c(0.2,.4,.4,.4))
                         
plot_grid(plot_chr21, plot_chrX)
```

