---
title: "CIN"
author: "Alicia Pliego"
output: 
  html_document:
    toc: true
    toc_float: true
    toc_collapsed: true
    toc_depth: 3
    number_sections: true
    theme: sandstone
---
<style>
body {
text-align: justify}
</style>

# Chromosomal Instability

Genomic instability, one of the hallmarks of cancer, is measured in many forms such as chromosomal instability, microsatellite instability, and instability characterized by increased frequency of base-pair mutations (Bakhoum and Cantley, 2018; Pikor et al., 2013; Negrini et al., 2010). Particularly, chromosomal instability (CIN) is associated with cancer progression, tumor immunity, and inflammation (Pikor et al., 2013; Bach et al., 2019). Recently, CIN has been shown to contribute to other diseases than cancer, including neurodegenerative diseases (Hou et al., 2017; Yurov et al., 2019).

```{r setup, include=FALSE, echo=FALSE, message=FALSE}
library(CINmetrics)
library(TCGAbiolinks)
library(heatmaply)
library(GenomicRanges)
library(AnnotationHub)
library(pd.genomewidesnp.6)
library(rtracklayer)
library(biovizBase) #needed for stain information
library(CINdex)
library(IRanges)
library(TxDb.Hsapiens.UCSC.hg18.knownGene)
library(Homo.sapiens)
```

## Genomic Instability meassurements

### Total aberration index (tai) and Modified Total Aberration Index (taiModified)

Total Aberration Index (TAI) was proposed by (Baumbusch et al., 2013) to measure the genomic aberrations in serous ovarian cancers. TAI calculates absolute area under the curve for a copy number segment profile generated by piecewise constant fitting (PCF) algorithm. Biologically, TAI can be interpreted as absolute deviation from the normal copy number state averaged over all genomic locations. TAI provides a numerical measure in terms of both prevalence as well as the genomic size of copy number variations in tumors. One of the limitations of TAI is that since it was designed for studying advance stage ovarian tumors and short aberrations found in the early tumors have low impact on TAI. Therefore, TAI should be used to study global scale genomic disorganization that most likely occur in late stage tumors.

tai implemented in CINmetrics takes into account only those sample values that are in aberrant copy number state, i.e. has a mean segment values of less than or equal to −0.2 and greater than or equal to +0.2.

### Copy number abnormality (cna) and Counting number of break points (countingBreakPoints)

Copy Number Abnormality (CNA) was developed by (Davison et al., 2014) for studying aneuploidy in superficial gastroesophageal adenocarcinoma. An individual CNA is defined as the segment with copy number outside the predefined range of 1.7 to 2.3 with the score of 2 indicating no loss or gain (assuming that the tumor is diploid) as determined by Partek segmentation algorithm. Total CNA for the sample can thus be defined as total number of individual CNAs. CNA can be thought of as a measure of segmental aneuploidy. cna implemented in CINmetrics is similar except we define individual CNA as the segment with copy number less than equal to −0.2 and greater than equal to +0.2 with segment mean of 0 indicating no loss or gain. We chose ±0.2 as a conservative cutoff for TCGA data as described in (Laddha et al., 2014). The users can modify the cutoff by modifying segmentMean parameter.

### Counting altered base segments (countingBaseSegments) and Fraction of the genome altered (fga)

Counting altered base segments and fraction of the genome altered are modified implementation of the Genome Instability Index (GII) as described in (Chin et al., 2007). The GII was computed in two different ways, both based on calculating common regions of alteration (CRA) and both approaches showed high concordance.

fga is based on identifying common regions of alterations as fraction of the genome altered. Therefore, the fga values are normalized by dividing it by the length of the genome covered. countingBaseSegments on the other hand calculates the common regions of alteration.

### Results CINmetrics

All the metrics were computed for TCGASARC, TCGA-OV, TCGA-CRC and log10 scaled in order to compare all the meassurements between them.

#### TCGA-SARC
```{r , include=FALSE, echo=FALSE, message=FALSE}
query.maskCNV.hg39.SARC <- GDCquery(project = "TCGA-SARC",
              data.category = "Copy Number Variation",
              data.type = "Masked Copy Number Segment", legacy=FALSE)
GDCdownload(query = query.maskCNV.hg39.SARC)
maskCNV.SARC <- GDCprepare(query = query.maskCNV.hg39.SARC, summarizedExperiment = FALSE)
maskCNV.SARC <- data.frame(maskCNV.SARC, stringsAsFactors = FALSE)
#tai.test <- tai(cnvData = maskCNV.SARC)
```



```{r , include=FALSE, echo=FALSE, message=FALSE}
#cinmetrics_test_sarc <- CINmetrics(cnvData = maskCNV.SARC)
#head(cinmetrics_test_sarc)
#write.table(cinmetrics_test_sarc, "/Users/aliciapliego/Projects/HRDsarcoma/data/cinmetrics_sarcoma.tsv", sep="\t", quote = FALSE, row.names = F)
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_sarc_anno <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc.tsv', sep = '\t')
cinmetrics_sarc_anno
cinmetrics_sarc_anno<- cinmetrics_sarc_anno[!duplicated(cinmetrics_sarc_anno), ]
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_sarc_anno_2 <- cinmetrics_sarc_anno 
rownames(cinmetrics_sarc_anno_2) -> cinmetrics_sarc_anno_2$sample_id
cinmetrics_sarc_anno_2$sample_id <- NULL
cinmetrics_sarc_anno_2$Tumor_type <- NULL
cinmetrics_sarc_anno_2$Sample.Type <- NULL
cinmetrics_sarc_anno_2$HRD34 <- NULL
cinmetrics_sarc_log10 <- log10(cinmetrics_sarc_anno_2)
```


##### Heatmap of the different scores 

 tai does not capture this global pattern of difference between normal and tumour samples. tai is best suited for late stage cancers, thus should be used as a measure for studying overall genomic disorganization in individual patients with advanced tumours and not as a measure of genomic instability comparison between normal and tumour samples.
 
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_sarc_log10),
  #xlab = "Features", 
  #ylab = "Features"
)
```

````{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_sarc_log10) <- cinmetrics_sarc_anno$sample_id
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cinmetrics_sarc_log10, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 200, 
    limits = c(0, 500)
  ),
  showticklabels = c(TRUE, FALSE),
  scale = "column",
  main = "TCGA-SARC CINmetrics",
  row_side_colors = cinmetrics_sarc_anno[, c("Tumor_type", "Sample.Type", "HRD34")]
  )
```


##### SINOVIAL SARCOMA (SS) Heatmap of the different scores 

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_sarc_anno <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc_SS.tsv', sep = '\t')
cinmetrics_sarc_anno
cinmetrics_sarc_anno<- cinmetrics_sarc_anno[!duplicated(cinmetrics_sarc_anno), ]
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_sarc_anno_2 <- cinmetrics_sarc_anno 
rownames(cinmetrics_sarc_anno_2) -> cinmetrics_sarc_anno_2$sample_id
cinmetrics_sarc_anno_2$sample_id <- NULL
cinmetrics_sarc_anno_2$Tumor_type <- NULL
cinmetrics_sarc_anno_2$Sample.Type <- NULL
cinmetrics_sarc_anno_2$HRD34 <- NULL
cinmetrics_sarc_log10 <- log10(cinmetrics_sarc_anno_2)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_sarc_log10),
  #xlab = "Features", 
  #ylab = "Features"
)
```

````{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_sarc_log10) <- cinmetrics_sarc_anno$sample_id
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cinmetrics_sarc_log10, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 200, 
    limits = c(0, 500)
  ),
  showticklabels = c(TRUE, FALSE),
  scale = "column",
  main = "Sinovial Sarcoma (SS) TCGA-SARC CINmetrics",
  row_side_colors = cinmetrics_sarc_anno[, c( "Sample.Type", "HRD34")]
  )
```

##### UPS Heatmap of the different scores 

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_sarc_anno <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc_UPS.tsv', sep = '\t')
cinmetrics_sarc_anno
cinmetrics_sarc_anno<- cinmetrics_sarc_anno[!duplicated(cinmetrics_sarc_anno), ]
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_sarc_anno_2 <- cinmetrics_sarc_anno 
rownames(cinmetrics_sarc_anno_2) -> cinmetrics_sarc_anno_2$sample_id
cinmetrics_sarc_anno_2$sample_id <- NULL
cinmetrics_sarc_anno_2$Tumor_type <- NULL
cinmetrics_sarc_anno_2$Sample.Type <- NULL
cinmetrics_sarc_anno_2$HRD34 <- NULL
cinmetrics_sarc_log10 <- log10(cinmetrics_sarc_anno_2)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_sarc_log10),
  #xlab = "Features", 
  #ylab = "Features"
)
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_sarc_log10) <- cinmetrics_sarc_anno$sample_id
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cinmetrics_sarc_log10, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 200, 
    limits = c(0, 500)
  ),
  showticklabels = c(TRUE, FALSE),
  scale = "column",
  main = "UPS TCGA-SARC CINmetrics",
  row_side_colors = cinmetrics_sarc_anno[, c( "Sample.Type", "HRD34")]
  )
```

##### MFS Heatmap of the different scores 

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_sarc_anno <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc_MFS.tsv', sep = '\t')
cinmetrics_sarc_anno
cinmetrics_sarc_anno<- cinmetrics_sarc_anno[!duplicated(cinmetrics_sarc_anno), ]
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_sarc_anno_2 <- cinmetrics_sarc_anno 
rownames(cinmetrics_sarc_anno_2) -> cinmetrics_sarc_anno_2$sample_id
cinmetrics_sarc_anno_2$sample_id <- NULL
cinmetrics_sarc_anno_2$Tumor_type <- NULL
cinmetrics_sarc_anno_2$Sample.Type <- NULL
cinmetrics_sarc_anno_2$HRD34 <- NULL
cinmetrics_sarc_log10 <- log10(cinmetrics_sarc_anno_2)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_sarc_log10),
  #xlab = "Features", 
  #ylab = "Features"
)
```

````{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_sarc_log10) <- cinmetrics_sarc_anno$sample_id
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cinmetrics_sarc_log10, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 200, 
    limits = c(0, 500)
  ),
  showticklabels = c(TRUE, FALSE),
  scale = "column",
  main = "MFS TCGA-SARC CINmetrics",
  row_side_colors = cinmetrics_sarc_anno[, c( "Sample.Type", "HRD34")]
  )
```

##### ddLPS Heatmap of the different scores 

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_sarc_anno <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc_ddLPS.tsv', sep = '\t')
cinmetrics_sarc_anno
cinmetrics_sarc_anno<- cinmetrics_sarc_anno[!duplicated(cinmetrics_sarc_anno), ]
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_sarc_anno_2 <- cinmetrics_sarc_anno 
rownames(cinmetrics_sarc_anno_2) -> cinmetrics_sarc_anno_2$sample_id
cinmetrics_sarc_anno_2$sample_id <- NULL
cinmetrics_sarc_anno_2$Tumor_type <- NULL
cinmetrics_sarc_anno_2$Sample.Type <- NULL
cinmetrics_sarc_anno_2$HRD34 <- NULL
cinmetrics_sarc_log10 <- log10(cinmetrics_sarc_anno_2)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_sarc_log10),
  #xlab = "Features", 
  #ylab = "Features"
)
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_sarc_log10) <- cinmetrics_sarc_anno$sample_id
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cinmetrics_sarc_log10, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 200, 
    limits = c(0, 500)
  ),
  showticklabels = c(TRUE, FALSE),
  scale = "column",
  main = "ddLPS TCGA-SARC CINmetrics",
  row_side_colors = cinmetrics_sarc_anno[, c( "Sample.Type", "HRD34")]
  )
```

##### LMS Heatmap of the different scores 

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_sarc_anno <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc_LMS.tsv', sep = '\t')
cinmetrics_sarc_anno
cinmetrics_sarc_anno<- cinmetrics_sarc_anno[!duplicated(cinmetrics_sarc_anno), ]
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_sarc_anno_2 <- cinmetrics_sarc_anno 
rownames(cinmetrics_sarc_anno_2) -> cinmetrics_sarc_anno_2$sample_id
cinmetrics_sarc_anno_2$sample_id <- NULL
cinmetrics_sarc_anno_2$Tumor_type <- NULL
cinmetrics_sarc_anno_2$Sample.Type <- NULL
cinmetrics_sarc_anno_2$HRD34 <- NULL
cinmetrics_sarc_log10 <- log10(cinmetrics_sarc_anno_2)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_sarc_log10),
  #xlab = "Features", 
  #ylab = "Features"
)
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_sarc_log10) <- cinmetrics_sarc_anno$sample_id
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cinmetrics_sarc_log10, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 200, 
    limits = c(0, 500)
  ),
  showticklabels = c(TRUE, FALSE),
  scale = "column",
  main = "LMS TCGA-SARC CINmetrics",
  row_side_colors = cinmetrics_sarc_anno[, c( "Sample.Type", "HRD34")]
  )
```

##### uLMS Heatmap of the different scores 

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_sarc_anno <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc_uLMS.tsv', sep = '\t')
cinmetrics_sarc_anno
cinmetrics_sarc_anno<- cinmetrics_sarc_anno[!duplicated(cinmetrics_sarc_anno), ]
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_sarc_anno_2 <- cinmetrics_sarc_anno 
rownames(cinmetrics_sarc_anno_2) -> cinmetrics_sarc_anno_2$sample_id
cinmetrics_sarc_anno_2$sample_id <- NULL
cinmetrics_sarc_anno_2$Tumor_type <- NULL
cinmetrics_sarc_anno_2$Sample.Type <- NULL
cinmetrics_sarc_anno_2$HRD34 <- NULL
cinmetrics_sarc_log10 <- log10(cinmetrics_sarc_anno_2)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_sarc_log10),
  #xlab = "Features", 
  #ylab = "Features"
)
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_sarc_log10) <- cinmetrics_sarc_anno$sample_id
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cinmetrics_sarc_log10, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 200, 
    limits = c(0, 500)
  ),
  showticklabels = c(TRUE, FALSE),
  scale = "column",
  main = "uLMS TCGA-SARC CINmetrics",
  row_side_colors = cinmetrics_sarc_anno[, c( "Sample.Type", "HRD34")]
  )
```

##### MPNST Heatmap of the different scores 

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_sarc_anno <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc_MPNST.tsv', sep = '\t')
cinmetrics_sarc_anno
cinmetrics_sarc_anno<- cinmetrics_sarc_anno[!duplicated(cinmetrics_sarc_anno), ]
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_sarc_anno_2 <- cinmetrics_sarc_anno 
rownames(cinmetrics_sarc_anno_2) -> cinmetrics_sarc_anno_2$sample_id
cinmetrics_sarc_anno_2$sample_id <- NULL
cinmetrics_sarc_anno_2$Tumor_type <- NULL
cinmetrics_sarc_anno_2$Sample.Type <- NULL
cinmetrics_sarc_anno_2$HRD34 <- NULL
cinmetrics_sarc_log10 <- log10(cinmetrics_sarc_anno_2)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_sarc_log10),
  #xlab = "Features", 
  #ylab = "Features"
)
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_sarc_log10) <- cinmetrics_sarc_anno$sample_id
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cinmetrics_sarc_log10, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 200, 
    limits = c(0, 500)
  ),
  showticklabels = c(TRUE, FALSE),
  scale = "column",
  main = "MPNST TCGA-SARC CINmetrics",
  row_side_colors = cinmetrics_sarc_anno[, c( "Sample.Type", "HRD34")]
  )
```

##### DT Heatmap of the different scores 

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_sarc_anno <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc_DT.tsv', sep = '\t')
cinmetrics_sarc_anno
cinmetrics_sarc_anno<- cinmetrics_sarc_anno[!duplicated(cinmetrics_sarc_anno), ]
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_sarc_anno_2 <- cinmetrics_sarc_anno 
rownames(cinmetrics_sarc_anno_2) -> cinmetrics_sarc_anno_2$sample_id
cinmetrics_sarc_anno_2$sample_id <- NULL
cinmetrics_sarc_anno_2$Tumor_type <- NULL
cinmetrics_sarc_anno_2$Sample.Type <- NULL
cinmetrics_sarc_anno_2$HRD34 <- NULL
cinmetrics_sarc_log10 <- log10(cinmetrics_sarc_anno_2)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_sarc_log10),
  #xlab = "Features", 
  #ylab = "Features"
)
```

````{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_sarc_log10) <- cinmetrics_sarc_anno$sample_id
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cinmetrics_sarc_log10, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 200, 
    limits = c(0, 500)
  ),
  showticklabels = c(TRUE, FALSE),
  scale = "column",
  main = "DT TCGA-SARC CINmetrics",
  row_side_colors = cinmetrics_sarc_anno[, c( "Sample.Type", "HRD34")]
  )
```

#### Correlation cinmetrics and HRD
```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_corr <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc_correlation_hrd.tsv', sep = '\t')
cinmetrics_corr
cinmetrics_corr<- cinmetrics_corr[!duplicated(cinmetrics_corr$sample_id), ]
```

````{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_corr_2 <- cinmetrics_corr 
rownames(cinmetrics_corr_2) -> cinmetrics_corr_2$sample_id
cinmetrics_corr_2$sample_id <- NULL
cinmetrics_corr_2$Tumor_type <- NULL
cinmetrics_corr_2$Sample.Type <- NULL
cinmetrics_corr_2$HRD34 <- NULL
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_corr_2) <- cinmetrics_corr$sample_id
```


```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_corr_2),
  #xlab = "Features", 
  #ylab = "Features"
)
```


```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
library(plotly)

fig <- plot_ly(
  cinmetrics_corr_2, x = ~HRDscore, y = ~break_points,
  # Hover text:
  text = ~rownames(cinmetrics_corr_2),
  color = ~HRDscore, size = ~break_points
) %>% layout(title = 'HRD34 vs Breakpoints')
fig
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig <- plot_ly(
  cinmetrics_corr_2, x = ~HRDscore, y = ~fga,
  # Hover text:
  text = ~rownames(cinmetrics_corr_2),
  color = ~HRDscore, size = ~break_points
) %>% layout(title = 'HRD34 vs fga')
fig
```


```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig <- plot_ly(
  cinmetrics_corr_2, x = ~HRDscore, y = ~cna,
  # Hover text:
  text = ~rownames(cinmetrics_corr_2),
  color = ~HRDscore, size = ~break_points
) %>% layout(title = 'HRD34 vs Cna')
fig
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig <- plot_ly(
  cinmetrics_corr_2, x = ~HRDscore, y = ~base_segments,
  # Hover text:
  text = ~rownames(cinmetrics_corr_2),
  color = ~HRDscore, size = ~break_points
) %>% layout(title = 'HRD34 vs Base_segments')
fig
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
library(stats)
library(tsne)
tsne = tsne(cinmetrics_corr_2, initial_dims = 2, k =2) 
tsne <- data.frame(tsne) 
```

#### CINmetrics tSNE plot

tSNE plot showing the clustering of normal and tumor tissue including HRD34 and the CINmetrics signatures.
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
pdb <- cbind(tsne,cinmetrics_corr$Sample.Type)
options(warn = -1)
fig <-  plot_ly(data = pdb ,x =  ~X1, y = ~X2, type = 'scatter', mode = 'markers', split = ~cinmetrics_corr$Sample.Type)

fig <- fig %>%
  layout(
    plot_bgcolor = "#e5ecf6"
  )

fig
```


#### CONCLUSION CINmetrics 

There is a correlation between HRD34 and the CINmetric signatures except for tai, which is stated by the cinmetrics publication that is a marker of normal or tumors in an earlier stage. In some types of cancer there seems to be a correlation between low HRD and lower breakpoints and cna, however the small data set limits this conclusions and more statistics are needed.

## CINdex

### CINdex at chromosome level
To mathematically and quantitatively describe these alternations we first locate
their genomic positions and measure their ranges. Such algorithms are referred to
as segmentation algorithms.

I run CINDex in the masked copy number segments from TCGA-SARC downloaded from GDC data commons. 
Masked copy number segments repport the segment mean which is the log2(copy-number/ 2). However, CINdex requires the CNV values. Therefore, in order to obtain the copy number value: 

$$
CNV = 2*2^{mean value}

$$
Settings:  CIN for a treshold gain of 2.25, threshold loss of 1.75, unnormalized (indicated by V.def=3), showing overall (gains and losses) CIN (indicated by V.mode=“sum’).

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Masked copy number data downloaded from TCGA has the mean normalized values, this is the log2(copy-number/ 2), meaning that the values equal to 0 are diploid, whilst the the ones with positive values are amplifies and below 0 are losses. 
maskCNV.SARC$norm <- 2*(2^maskCNV.SARC$Segment_Mean)
maskCNV.SARC$norm <- round(maskCNV.SARC$norm, 4)
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
#function to convert Copy number data into GRanges object
convert.to.gr.cnv <- function(cnv2) {
cnv2$Chromosome <- cnv2$Chromosome
# subset out SNPs with missing location
cnv2 <- cnv2[!(is.na(cnv2$Chromosome) | is.na(cnv2$Start)),]
# convert strand info to +,-,*

cnv2$strand <- c( "*")
#convert into GRanges object
cnv2.gr <- GRanges(cnv2$Chromosome, IRanges(cnv2$Start,cnv2$End),cnv2$strand, value = cnv2$norm)
return(cnv2.gr)
}
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
v1 <- list()
for(un1 in unique(maskCNV.SARC$Sample)){
  tmp <- maskCNV.SARC[maskCNV.SARC$Sample == un1, ]
  tmp <- tmp[!grepl("X", tmp$Chromosome),]
  tmp <- tmp[!grepl("Y", tmp$Chromosome),]
  cnv2gr <- convert.to.gr.cnv(cnv2 = tmp)
  
  #Subset only the probes that are in autosomes
  
  v1<- c(v1, cnv2gr)
}

cnvgr.tcga.sarc <- GRangesList(v1)
names(cnvgr.tcga.sarc)<- unique(maskCNV.SARC$Sample)
```

```{r cindex sum, include=FALSE, echo=FALSE, message=FALSE}
# Run CINdex by chromosome 
run.cin.chr(grl.seg = cnvgr.tcga.sarc, thr.gain=2.25, thr.loss=1.75, V.def=3, V.mode="sum")
```

```{r cindex amp, include=FALSE, echo=FALSE, message=FALSE}
# Run CINdex by chromosome 
run.cin.chr(grl.seg = cnvgr.tcga.sarc, thr.gain=2.25, thr.loss=1.75, V.def=3, V.mode="amp")
```

```{r cindex del , include=FALSE, echo=FALSE, message=FALSE}
# Run CINdex by chromosome 
run.cin.chr(grl.seg = cnvgr.tcga.sarc, thr.gain=2.25, thr.loss=1.75, V.def=3, V.mode="del")
```

```{r save cindex results, include=FALSE, echo=FALSE, message=FALSE}
#cin.sarc.results <- load("/Users/aliciapliego/Projects/HRDsarcoma/output_chr_cin/dataMatrix_2.25_1.75_unnormalized_sum.RData")
#cin.sarc.results.df <- dataMatrix
#cin.sarc.results.df <- t(cin.sarc.results.df)
#write.table(cin.sarc.results.df, "/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_sarc_results.tsv", sep="\t", quote = FALSE, row.names = T)

#cin.sarc.amp.results <- load("/Users/aliciapliego/Projects/HRDsarcoma/output_chr_cin/dataMatrix_2.25_1.75_unnormalized_amp.RData")
#cin.sarc.amp.results.df <- dataMatrix
#cin.sarc.amp.results.df <- t(cin.sarc.amp.results.df)
#write.table(cin.sarc.amp.results.df, "/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_sarc_amp_results.tsv", sep="\t", quote = FALSE, row.names = T)

cin.sarc.del.results <- load("/Users/aliciapliego/Projects/HRDsarcoma/output_chr_cin/dataMatrix_2.25_1.75_unnormalized_del.RData")
#cin.sarc.del.results.df <- dataMatrix
#cin.sarc.del.results.df <- t(cin.sarc.del.results.df)
#write.table(cin.sarc.del.results.df, "/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_sarc_del_results.tsv", sep="\t", quote = FALSE, row.names = T)
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Edit results in python and annotate them and add files back here
cindex_results_sum_sarc <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc.tsv', sep = '\t', header = TRUE)

rownames(cindex_results_sum_sarc) <- cindex_results_sum_sarc$Sample.ID
cindex_results_sum_sarc$Sample.ID <- NULL

cindex_results_sarc_meta <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_metadata.tsv', sep = '\t', header = TRUE)


```

```{r heatmap cindex all, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cindex_results_sum_sarc, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "CINdex TCGA",
  row_side_colors = cindex_results_sarc_meta[, c( "Sample.Type", "Tumor_type", "HRD34")]
                                             
  )
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cindex_results_sum_sarc_bp <- read.table('data/cindex_results_sarc_ptumor.tsv', sep = '\t', header = TRUE)
cindex_results_sum_sarc_meta_bp <- read.table('data/cindex_results_sarc_metadata_ptumor.tsv', sep = '\t', header = TRUE)
cindex_results_sum_sarc_bp$HRD34 <- cindex_results_sum_sarc_meta_bp$HRD34
cindex_results_sum_sarc_bp$Tumor_type <- cindex_results_sum_sarc_meta_bp$Tumor_type
```

```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
cindex_results_sum_sarc_bp$sums <- rowSums(cindex_results_sum_sarc_bp[2:23])

fig1 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~sums, color = ~HRD34, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by HRD"), boxmode= 'group',
     yaxis = list())
fig1
```


```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig1 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr1, color = ~HRD34, type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig1
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}
fig2 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr2, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))

fig2
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}
fig3 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr3, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig3
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}
fig4 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr4, color = ~HRD34, type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig4
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}
fig5 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr5, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig5
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}
fig6 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr6, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig6
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}
fig7 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr7, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig7
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}
fig8 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr8, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig8

```


```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig9 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr9, color = ~HRD34, type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig9
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}
fig10 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr10, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig10
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}
fig11 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr11, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig11
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig12 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr12, color = ~HRD34, type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig12
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig13 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr13, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig13
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig14 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr14, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig14

```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig15 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr15, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig15
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig16 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr16, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))

fig16
```


```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig17 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr17, color = ~HRD34, type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig17
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig18 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr18, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig18
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig19 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr19, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig19
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig20 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr20, color = ~HRD34, type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig20
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig21 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr21, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig21
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig22 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr22, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig22
```



Summary:

 * chr1: CIN is significantly higher in UPS HRD high and LMS HRD high.  
 
 * chr2: CIN is significantly higher in MFS MPNST and UPS in HRD high groups.  
 
 * chr3: CIN is significantly higher in MFS, LMS and UPS in HRD high groups.  
 
 * chr4: CIN is significantly higher in MFS, uLMS and UPS and MPNST in HRD high groups.  
 
 * chr5: CIN is significantly higher in LMS, UPS, uLMS, MFS and MPNST in HRD high groups.  
 
 * chr6: CIN is significantly higher in MFS, and UPS in HRD high groups.  
 
 * chr7: CIN is significantly higher in MPNST in HRD high groups.  
 
 * chr8: CIN is significantly higher in MPNST, MFS, LMS and UPS in HRD high groups.  
 
 * chr9: CIN is significantly higher in UPS HRD high and LMS HRD high.  
 
 * chr10: CIN is significantly higher in uLMS in HRD high groups.  
 
 * chr11: CIN is significantly higher in MFS, LMS, MPNST and UPS in HRD high groups.  
 
 * chr12: CIN is significantly higher in ddLPS and MPNST in HRD high groups. MFS patient TCGA-DX-AB2S-01A with low HRD shows a very high number of CIN in chromosome 12.  
 
 * chr13: CIN is significantly higher in UPSand ddLPS in HRD high groups.  
 
 * chr14: CIN is significantly higher in MPNST in HRD high groups.  
 
 * chr15: CIN is significantly higher in MPNST in HRD high groups.  
 
 * chr16: CIN is significantly higher in MPNST, MFS, LMS and UPS in HRD high groups.  
 
 * chr17: CIN is significantly higher in LMS, ddLPS and UPS in HRD high groups.  
 
 * chr18: CIN is significantly higher in MFS, MPNST, uLMS and UPS in HRD high groups.  
 
 * chr19: CIN is significantly higher in UPS  in HRD high groups.  
 
 * chr20: CIN is significantly higher in MFS and UPS in HRD high groups.
 * chr21: -
 * chr22: -

In general we can see how HRD high groups show high CIN in individual chromosomes.

### CINdex MFS
```{r cindex mfs, include=FALSE, echo=FALSE, message=FALSE}
cindex_results_sarc_MFS <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_MFS.tsv', sep = '\t', header = TRUE)

rownames(cindex_results_sarc_MFS) <- cindex_results_sarc_MFS$Sample.ID
cindex_results_sarc_MFS$Sample.ID <- NULL

cindex_results_sarc_MFS_meta <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_metadata_MFS.tsv', sep = '\t', header = TRUE)
```

```{r heatmap cindex mfs, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cindex_results_sarc_MFS, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "CINdex TCGA MFS",
  row_side_colors = cindex_results_sarc_MFS_meta[, c("Tumor_type", "HRD34")]
                                             
  )
```

### CINdex ddLPS
```{r cindex ddlps, include=FALSE, echo=FALSE, message=FALSE}
cindex_results_sarc_ddLPS <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_ddLPS.tsv', sep = '\t', header = TRUE)

rownames(cindex_results_sarc_ddLPS) <- cindex_results_sarc_ddLPS$Sample.ID
cindex_results_sarc_ddLPS$Sample.ID <- NULL

cindex_results_sarc_ddLPS_meta <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_metadata_ddLPS.tsv', sep = '\t', header = TRUE)
```

```{r heatmap cindex ddlps, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cindex_results_sarc_ddLPS, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "CINdex TCGA ddLPS",
  row_side_colors = cindex_results_sarc_ddLPS_meta[, c("Tumor_type", "HRD34")]
                                             
  )
```

### CINdex LMS
```{r cindex lms, include=FALSE, echo=FALSE, message=FALSE}
cindex_results_sarc_LMS <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_LMS.tsv', sep = '\t', header = TRUE)

rownames(cindex_results_sarc_LMS) <- cindex_results_sarc_LMS$Sample.ID
cindex_results_sarc_LMS$Sample.ID <- NULL

cindex_results_sarc_LMS_meta <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_metadata_LMS.tsv', sep = '\t', header = TRUE)
```

```{r heatmap cindex lms, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cindex_results_sarc_LMS, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "CINdex TCGA LMS",
  row_side_colors = cindex_results_sarc_LMS_meta[, c("Tumor_type", "HRD34")]
                                             
  )
```

### CINdex UPS
```{r cindex ups, include=FALSE, echo=FALSE, message=FALSE}
cindex_results_sarc_UPS <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_UPS.tsv', sep = '\t', header = TRUE)

rownames(cindex_results_sarc_UPS) <- cindex_results_sarc_UPS$Sample.ID
cindex_results_sarc_UPS$Sample.ID <- NULL

cindex_results_sarc_UPS_meta <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_metadata_UPS.tsv', sep = '\t', header = TRUE)
```

```{r heatmap cindex ups, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cindex_results_sarc_UPS, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "CINdex TCGA UPS",
  row_side_colors = cindex_results_sarc_UPS_meta[, c("Tumor_type", "HRD34")]
                                             
  )
```

### CINdex uLMS
```{r cindex ulms, include=FALSE, echo=FALSE, message=FALSE}
cindex_results_sarc_uLMS <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_uLMS.tsv', sep = '\t', header = TRUE)

rownames(cindex_results_sarc_uLMS) <- cindex_results_sarc_uLMS$Sample.ID
cindex_results_sarc_uLMS$Sample.ID <- NULL

cindex_results_sarc_uLMS_meta <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_metadata_uLMS.tsv', sep = '\t', header = TRUE)
```

```{r heatmap cindex ulms, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cindex_results_sarc_uLMS, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "CINdex TCGA uLMS",
  row_side_colors = cindex_results_sarc_uLMS_meta[, c("Tumor_type", "HRD34")]
                                             
  )
```

### CINdex SS
```{r cindex ss, include=FALSE, echo=FALSE, message=FALSE}
cindex_results_sarc_SS <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_SS.tsv', sep = '\t', header = TRUE)

rownames(cindex_results_sarc_SS) <- cindex_results_sarc_SS$Sample.ID
cindex_results_sarc_SS$Sample.ID <- NULL

cindex_results_sarc_SS_meta <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_metadata_SS.tsv', sep = '\t', header = TRUE)
```

```{r heatmap cindex ss, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cindex_results_sarc_SS, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "CINdex TCGA SS",
  row_side_colors = cindex_results_sarc_SS_meta[, c("Tumor_type", "HRD34")]
                                             
  )
```

### CINdex MPNST
```{r cindex mpnst, include=FALSE, echo=FALSE, message=FALSE}
cindex_results_sarc_MPNST <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_MPNST.tsv', sep = '\t', header = TRUE)

rownames(cindex_results_sarc_MPNST) <- cindex_results_sarc_MPNST$Sample.ID
cindex_results_sarc_MPNST$Sample.ID <- NULL

cindex_results_sarc_MPNST_meta <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_metadata_MPNST.tsv', sep = '\t', header = TRUE)
```

```{r heatmap cindex mpnst, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cindex_results_sarc_MPNST, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "CINdex TCGA MPNST",
  row_side_colors = cindex_results_sarc_MPNST_meta[, c("Tumor_type", "HRD34")]
                                             
  )
```

### CINdex DT
```{r cindex dt, include=FALSE, echo=FALSE, message=FALSE}
cindex_results_sarc_DT <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_DT.tsv', sep = '\t', header = TRUE)

rownames(cindex_results_sarc_DT) <- cindex_results_sarc_DT$Sample.ID
cindex_results_sarc_DT$Sample.ID <- NULL

cindex_results_sarc_DT_meta <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_metadata_DT.tsv', sep = '\t', header = TRUE)
```

```{r heatmap cindex dt, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cindex_results_sarc_DT, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "CINdex TCGA DT",
  row_side_colors = cindex_results_sarc_DT_meta[, c("Tumor_type", "HRD34")]
                                             
  )
```

### CINdex CINmetrics HRDness correlation

```{r cindex cinmetrics cindex correlation, include=FALSE, echo=FALSE, message=FALSE}
cindex_cinmetrix_corr<- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_cinmetrix_corr.tsv', sep = '\t', header = TRUE)
cindex_cinmetrix_corr<- cindex_cinmetrix_corr[!duplicated(cindex_cinmetrix_corr), ]
rownames(cindex_cinmetrix_corr) <- cindex_cinmetrix_corr$Sample.ID
cindex_cinmetrix_corr$Sample.ID <- NULL
cindex_cinmetrix_corr_log10 <- log10(cindex_cinmetrix_corr)
```

```{r heatmap cinmetrics cindex correlation , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cindex_cinmetrix_corr),
  #xlab = "Features", 
  #ylab = "Features"
)
```

There is a High correlation between CINdex and the cna and break points, but there is also certain correlation with fga and base_segments.
#### All meassurements 
```{r all metrics, include=FALSE, echo=FALSE, message=FALSE}
#Filtered only by primary tumor 
all_metrics <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_cinmetrix_hrdscore_sarc.tsv', sep = '\t', header = TRUE)

```
##### fga in TCGA-SARC
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig <- plot_ly(all_metrics, y = ~fga, color = ~Tumor_type, type = "box",  text = all_metrics$Sample.ID, notched=TRUE) %>%
layout(
     xaxis = list(title = "",
     categoryorder = "array",
     categoryarray = c(
                         'UPS',
                         'MFS',
                         'uLMS',
                         'MPNST',
                         'LMS',
                         'ddLPS',
                         'SS',
                         'DT'))
)

fig
```

##### break_points in TCGA-SARC
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig <- plot_ly(all_metrics, y = ~break_points, color = ~Tumor_type, type = "box",  text = all_metrics$Sample.ID, notched=TRUE) %>%
layout(
     xaxis = list(title = "",
     categoryorder = "array",
     categoryarray = c(
                         'UPS',
                         'MFS',
                         'uLMS',
                         'MPNST',
                         'LMS',
                         'ddLPS',
                         'SS',
                         'DT'))
)

fig
```

##### cna in TCGA-SARC
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig <- plot_ly(all_metrics, y = ~cna, color = ~Tumor_type, type = "box",  text = all_metrics$Sample.ID, notched=TRUE) %>%
layout(
     xaxis = list(title = "",
     categoryorder = "array",
     categoryarray = c(
                         'UPS',
                         'MFS',
                         'uLMS',
                         'MPNST',
                         'LMS',
                         'ddLPS',
                         'SS',
                         'DT'))
)

fig
```

##### base_segments	 in TCGA-SARC
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig <- plot_ly(all_metrics, y = ~base_segments, color = ~Tumor_type, type = "box",  text = all_metrics$Sample.ID, notched=TRUE) %>%
layout(
     xaxis = list(title = "",
     categoryorder = "array",
     categoryarray = c(
                         'UPS',
                         'MFS',
                         'uLMS',
                         'MPNST',
                         'LMS',
                         'ddLPS',
                         'SS',
                         'DT'))
)

fig
```

##### CINdex	 in TCGA-SARC
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig <- plot_ly(all_metrics, y = ~CINdex_score, color = ~Tumor_type, type = "box",  text = all_metrics$Sample.ID, notched=TRUE) %>%
layout(
     xaxis = list(title = "",
     categoryorder = "array",
     categoryarray = c(
                         'UPS',
                         'MFS',
                         'uLMS',
                         'MPNST',
                         'LMS',
                         'ddLPS',
                         'SS',
                         'DT'))
)

fig
```

