---
title: "CIN"
author: "Alicia Pliego"
output: 
  html_document:
    toc: true
    toc_float: true
    toc_collapsed: true
    toc_depth: 3
    number_sections: true
    theme: sandstone
---
<style>
body {
text-align: justify}
</style>

# Chromosomal Instability

Genomic instability, one of the hallmarks of cancer, is measured in many forms such as chromosomal instability, microsatellite instability, and instability characterized by increased frequency of base-pair mutations (Bakhoum and Cantley, 2018; Pikor et al., 2013; Negrini et al., 2010). Particularly, chromosomal instability (CIN) is associated with cancer progression, tumor immunity, and inflammation (Pikor et al., 2013; Bach et al., 2019). Recently, CIN has been shown to contribute to other diseases than cancer, including neurodegenerative diseases (Hou et al., 2017; Yurov et al., 2019).

```{r setup, include=FALSE, echo=FALSE, message=FALSE}
library(CINmetrics)
library(TCGAbiolinks)
library(heatmaply)
library(GenomicRanges)
library(AnnotationHub)
library(pd.genomewidesnp.6)
library(rtracklayer)
library(biovizBase) #needed for stain information
library(CINdex)
library(IRanges)
library(TxDb.Hsapiens.UCSC.hg18.knownGene)
library(Homo.sapiens)
```

## Genomic Instability meassurements

### Total aberration index (tai) and Modified Total Aberration Index (taiModified)

Total Aberration Index (TAI) was proposed by (Baumbusch et al., 2013) to measure the genomic aberrations in serous ovarian cancers. TAI calculates absolute area under the curve for a copy number segment profile generated by piecewise constant fitting (PCF) algorithm. Biologically, TAI can be interpreted as absolute deviation from the normal copy number state averaged over all genomic locations. TAI provides a numerical measure in terms of both prevalence as well as the genomic size of copy number variations in tumors. One of the limitations of TAI is that since it was designed for studying advance stage ovarian tumors and short aberrations found in the early tumors have low impact on TAI. Therefore, TAI should be used to study global scale genomic disorganization that most likely occur in late stage tumors.

tai implemented in CINmetrics takes into account only those sample values that are in aberrant copy number state, i.e. has a mean segment values of less than or equal to −0.2 and greater than or equal to +0.2.

### Copy number abnormality (cna) and Counting number of break points (countingBreakPoints)

Copy Number Abnormality (CNA) was developed by (Davison et al., 2014) for studying aneuploidy in superficial gastroesophageal adenocarcinoma. An individual CNA is defined as the segment with copy number outside the predefined range of 1.7 to 2.3 with the score of 2 indicating no loss or gain (assuming that the tumor is diploid) as determined by Partek segmentation algorithm. Total CNA for the sample can thus be defined as total number of individual CNAs. CNA can be thought of as a measure of segmental aneuploidy. cna implemented in CINmetrics is similar except we define individual CNA as the segment with copy number less than equal to −0.2 and greater than equal to +0.2 with segment mean of 0 indicating no loss or gain. We chose ±0.2 as a conservative cutoff for TCGA data as described in (Laddha et al., 2014). The users can modify the cutoff by modifying segmentMean parameter.

### Counting altered base segments (countingBaseSegments) and Fraction of the genome altered (fga)

Counting altered base segments and fraction of the genome altered are modified implementation of the Genome Instability Index (GII) as described in (Chin et al., 2007). The GII was computed in two different ways, both based on calculating common regions of alteration (CRA) and both approaches showed high concordance.

fga is based on identifying common regions of alterations as fraction of the genome altered. Therefore, the fga values are normalized by dividing it by the length of the genome covered. countingBaseSegments on the other hand calculates the common regions of alteration.

### Results CINmetrics

All the metrics were computed for TCGASARC, TCGA-OV, TCGA-CRC and log10 scaled in order to compare all the meassurements between them.

#### TCGA-SARC
```{r , include=FALSE, echo=FALSE, message=FALSE}
query.maskCNV.hg39.SARC <- GDCquery(project = "TCGA-SARC",
              data.category = "Copy Number Variation",
              data.type = "Masked Copy Number Segment", legacy=FALSE)
GDCdownload(query = query.maskCNV.hg39.SARC)
maskCNV.SARC <- GDCprepare(query = query.maskCNV.hg39.SARC, summarizedExperiment = FALSE)
maskCNV.SARC <- data.frame(maskCNV.SARC, stringsAsFactors = FALSE)
#tai.test <- tai(cnvData = maskCNV.SARC)
```



```{r , include=FALSE, echo=FALSE, message=FALSE}
#cinmetrics_test_sarc <- CINmetrics(cnvData = maskCNV.SARC)
#head(cinmetrics_test_sarc)
#write.table(cinmetrics_test_sarc, "/Users/aliciapliego/Projects/HRDsarcoma/data/cinmetrics_sarcoma.tsv", sep="\t", quote = FALSE, row.names = F)
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_sarc_anno <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc.tsv', sep = '\t')
cinmetrics_sarc_anno
cinmetrics_sarc_anno<- cinmetrics_sarc_anno[!duplicated(cinmetrics_sarc_anno), ]
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_sarc_anno_2 <- cinmetrics_sarc_anno 
rownames(cinmetrics_sarc_anno_2) -> cinmetrics_sarc_anno_2$sample_id
cinmetrics_sarc_anno_2$sample_id <- NULL
cinmetrics_sarc_anno_2$Tumor_type <- NULL
cinmetrics_sarc_anno_2$Sample.Type <- NULL
cinmetrics_sarc_anno_2$HRD34 <- NULL
cinmetrics_sarc_log10 <- log10(cinmetrics_sarc_anno_2)
```


##### Heatmap of the different scores 

 tai does not capture this global pattern of difference between normal and tumour samples. tai is best suited for late stage cancers, thus should be used as a measure for studying overall genomic disorganization in individual patients with advanced tumours and not as a measure of genomic instability comparison between normal and tumour samples.
 
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_sarc_log10),
  #xlab = "Features", 
  #ylab = "Features"
)
```

````{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_sarc_log10) <- cinmetrics_sarc_anno$sample_id
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cinmetrics_sarc_log10, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 200, 
    limits = c(0, 500)
  ),
  showticklabels = c(TRUE, FALSE),
  scale = "column",
  main = "TCGA-SARC CINmetrics",
  row_side_colors = cinmetrics_sarc_anno[, c("Tumor_type", "Sample.Type", "HRD34")]
  )
```


##### SINOVIAL SARCOMA (SS) Heatmap of the different scores 

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_sarc_anno <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc_SS.tsv', sep = '\t')
cinmetrics_sarc_anno
cinmetrics_sarc_anno<- cinmetrics_sarc_anno[!duplicated(cinmetrics_sarc_anno), ]
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_sarc_anno_2 <- cinmetrics_sarc_anno 
rownames(cinmetrics_sarc_anno_2) -> cinmetrics_sarc_anno_2$sample_id
cinmetrics_sarc_anno_2$sample_id <- NULL
cinmetrics_sarc_anno_2$Tumor_type <- NULL
cinmetrics_sarc_anno_2$Sample.Type <- NULL
cinmetrics_sarc_anno_2$HRD34 <- NULL
cinmetrics_sarc_log10 <- log10(cinmetrics_sarc_anno_2)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_sarc_log10),
  #xlab = "Features", 
  #ylab = "Features"
)
```

````{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_sarc_log10) <- cinmetrics_sarc_anno$sample_id
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cinmetrics_sarc_log10, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 200, 
    limits = c(0, 500)
  ),
  showticklabels = c(TRUE, FALSE),
  scale = "column",
  main = "Sinovial Sarcoma (SS) TCGA-SARC CINmetrics",
  row_side_colors = cinmetrics_sarc_anno[, c( "Sample.Type", "HRD34")]
  )
```

##### UPS Heatmap of the different scores 

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_sarc_anno <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc_UPS.tsv', sep = '\t')
cinmetrics_sarc_anno
cinmetrics_sarc_anno<- cinmetrics_sarc_anno[!duplicated(cinmetrics_sarc_anno), ]
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_sarc_anno_2 <- cinmetrics_sarc_anno 
rownames(cinmetrics_sarc_anno_2) -> cinmetrics_sarc_anno_2$sample_id
cinmetrics_sarc_anno_2$sample_id <- NULL
cinmetrics_sarc_anno_2$Tumor_type <- NULL
cinmetrics_sarc_anno_2$Sample.Type <- NULL
cinmetrics_sarc_anno_2$HRD34 <- NULL
cinmetrics_sarc_log10 <- log10(cinmetrics_sarc_anno_2)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_sarc_log10),
  #xlab = "Features", 
  #ylab = "Features"
)
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_sarc_log10) <- cinmetrics_sarc_anno$sample_id
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cinmetrics_sarc_log10, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 200, 
    limits = c(0, 500)
  ),
  showticklabels = c(TRUE, FALSE),
  scale = "column",
  main = "UPS TCGA-SARC CINmetrics",
  row_side_colors = cinmetrics_sarc_anno[, c( "Sample.Type", "HRD34")]
  )
```

##### MFS Heatmap of the different scores 

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_sarc_anno <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc_MFS.tsv', sep = '\t')
cinmetrics_sarc_anno
cinmetrics_sarc_anno<- cinmetrics_sarc_anno[!duplicated(cinmetrics_sarc_anno), ]
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_sarc_anno_2 <- cinmetrics_sarc_anno 
rownames(cinmetrics_sarc_anno_2) -> cinmetrics_sarc_anno_2$sample_id
cinmetrics_sarc_anno_2$sample_id <- NULL
cinmetrics_sarc_anno_2$Tumor_type <- NULL
cinmetrics_sarc_anno_2$Sample.Type <- NULL
cinmetrics_sarc_anno_2$HRD34 <- NULL
cinmetrics_sarc_log10 <- log10(cinmetrics_sarc_anno_2)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_sarc_log10),
  #xlab = "Features", 
  #ylab = "Features"
)
```

````{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_sarc_log10) <- cinmetrics_sarc_anno$sample_id
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cinmetrics_sarc_log10, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 200, 
    limits = c(0, 500)
  ),
  showticklabels = c(TRUE, FALSE),
  scale = "column",
  main = "MFS TCGA-SARC CINmetrics",
  row_side_colors = cinmetrics_sarc_anno[, c( "Sample.Type", "HRD34")]
  )
```

##### ddLPS Heatmap of the different scores 

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_sarc_anno <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc_ddLPS.tsv', sep = '\t')
cinmetrics_sarc_anno
cinmetrics_sarc_anno<- cinmetrics_sarc_anno[!duplicated(cinmetrics_sarc_anno), ]
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_sarc_anno_2 <- cinmetrics_sarc_anno 
rownames(cinmetrics_sarc_anno_2) -> cinmetrics_sarc_anno_2$sample_id
cinmetrics_sarc_anno_2$sample_id <- NULL
cinmetrics_sarc_anno_2$Tumor_type <- NULL
cinmetrics_sarc_anno_2$Sample.Type <- NULL
cinmetrics_sarc_anno_2$HRD34 <- NULL
cinmetrics_sarc_log10 <- log10(cinmetrics_sarc_anno_2)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_sarc_log10),
  #xlab = "Features", 
  #ylab = "Features"
)
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_sarc_log10) <- cinmetrics_sarc_anno$sample_id
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cinmetrics_sarc_log10, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 200, 
    limits = c(0, 500)
  ),
  showticklabels = c(TRUE, FALSE),
  scale = "column",
  main = "ddLPS TCGA-SARC CINmetrics",
  row_side_colors = cinmetrics_sarc_anno[, c( "Sample.Type", "HRD34")]
  )
```

##### LMS Heatmap of the different scores 

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_sarc_anno <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc_LMS.tsv', sep = '\t')
cinmetrics_sarc_anno
cinmetrics_sarc_anno<- cinmetrics_sarc_anno[!duplicated(cinmetrics_sarc_anno), ]
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_sarc_anno_2 <- cinmetrics_sarc_anno 
rownames(cinmetrics_sarc_anno_2) -> cinmetrics_sarc_anno_2$sample_id
cinmetrics_sarc_anno_2$sample_id <- NULL
cinmetrics_sarc_anno_2$Tumor_type <- NULL
cinmetrics_sarc_anno_2$Sample.Type <- NULL
cinmetrics_sarc_anno_2$HRD34 <- NULL
cinmetrics_sarc_log10 <- log10(cinmetrics_sarc_anno_2)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_sarc_log10),
  #xlab = "Features", 
  #ylab = "Features"
)
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_sarc_log10) <- cinmetrics_sarc_anno$sample_id
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cinmetrics_sarc_log10, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 200, 
    limits = c(0, 500)
  ),
  showticklabels = c(TRUE, FALSE),
  scale = "column",
  main = "LMS TCGA-SARC CINmetrics",
  row_side_colors = cinmetrics_sarc_anno[, c( "Sample.Type", "HRD34")]
  )
```

##### uLMS Heatmap of the different scores 

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_sarc_anno <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc_uLMS.tsv', sep = '\t')
cinmetrics_sarc_anno
cinmetrics_sarc_anno<- cinmetrics_sarc_anno[!duplicated(cinmetrics_sarc_anno), ]
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_sarc_anno_2 <- cinmetrics_sarc_anno 
rownames(cinmetrics_sarc_anno_2) -> cinmetrics_sarc_anno_2$sample_id
cinmetrics_sarc_anno_2$sample_id <- NULL
cinmetrics_sarc_anno_2$Tumor_type <- NULL
cinmetrics_sarc_anno_2$Sample.Type <- NULL
cinmetrics_sarc_anno_2$HRD34 <- NULL
cinmetrics_sarc_log10 <- log10(cinmetrics_sarc_anno_2)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_sarc_log10),
  #xlab = "Features", 
  #ylab = "Features"
)
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_sarc_log10) <- cinmetrics_sarc_anno$sample_id
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cinmetrics_sarc_log10, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 200, 
    limits = c(0, 500)
  ),
  showticklabels = c(TRUE, FALSE),
  scale = "column",
  main = "uLMS TCGA-SARC CINmetrics",
  row_side_colors = cinmetrics_sarc_anno[, c( "Sample.Type", "HRD34")]
  )
```

##### MPNST Heatmap of the different scores 

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_sarc_anno <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc_MPNST.tsv', sep = '\t')
cinmetrics_sarc_anno
cinmetrics_sarc_anno<- cinmetrics_sarc_anno[!duplicated(cinmetrics_sarc_anno), ]
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_sarc_anno_2 <- cinmetrics_sarc_anno 
rownames(cinmetrics_sarc_anno_2) -> cinmetrics_sarc_anno_2$sample_id
cinmetrics_sarc_anno_2$sample_id <- NULL
cinmetrics_sarc_anno_2$Tumor_type <- NULL
cinmetrics_sarc_anno_2$Sample.Type <- NULL
cinmetrics_sarc_anno_2$HRD34 <- NULL
cinmetrics_sarc_log10 <- log10(cinmetrics_sarc_anno_2)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_sarc_log10),
  #xlab = "Features", 
  #ylab = "Features"
)
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_sarc_log10) <- cinmetrics_sarc_anno$sample_id
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cinmetrics_sarc_log10, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 200, 
    limits = c(0, 500)
  ),
  showticklabels = c(TRUE, FALSE),
  scale = "column",
  main = "MPNST TCGA-SARC CINmetrics",
  row_side_colors = cinmetrics_sarc_anno[, c( "Sample.Type", "HRD34")]
  )
```

##### DT Heatmap of the different scores 

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_sarc_anno <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc_DT.tsv', sep = '\t')
cinmetrics_sarc_anno
cinmetrics_sarc_anno<- cinmetrics_sarc_anno[!duplicated(cinmetrics_sarc_anno), ]
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_sarc_anno_2 <- cinmetrics_sarc_anno 
rownames(cinmetrics_sarc_anno_2) -> cinmetrics_sarc_anno_2$sample_id
cinmetrics_sarc_anno_2$sample_id <- NULL
cinmetrics_sarc_anno_2$Tumor_type <- NULL
cinmetrics_sarc_anno_2$Sample.Type <- NULL
cinmetrics_sarc_anno_2$HRD34 <- NULL
cinmetrics_sarc_log10 <- log10(cinmetrics_sarc_anno_2)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_sarc_log10),
  #xlab = "Features", 
  #ylab = "Features"
)
```

````{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_sarc_log10) <- cinmetrics_sarc_anno$sample_id
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cinmetrics_sarc_log10, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 200, 
    limits = c(0, 500)
  ),
  showticklabels = c(TRUE, FALSE),
  scale = "column",
  main = "DT TCGA-SARC CINmetrics",
  row_side_colors = cinmetrics_sarc_anno[, c( "Sample.Type", "HRD34")]
  )
```

#### Correlation cinmetrics and HRD
```{r , include=FALSE, echo=FALSE, message=FALSE}
#Load data from CINmetrics annotated in python
cinmetrics_corr <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/CINmetrics_annotated_sarc_correlation_hrd.tsv', sep = '\t')
cinmetrics_corr
cinmetrics_corr<- cinmetrics_corr[!duplicated(cinmetrics_corr$sample_id), ]
```

````{r , include=FALSE, echo=FALSE, message=FALSE}
cinmetrics_corr_2 <- cinmetrics_corr 
rownames(cinmetrics_corr_2) -> cinmetrics_corr_2$sample_id
cinmetrics_corr_2$sample_id <- NULL
cinmetrics_corr_2$Tumor_type <- NULL
cinmetrics_corr_2$Sample.Type <- NULL
cinmetrics_corr_2$HRD34 <- NULL
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
row.names(cinmetrics_corr_2) <- cinmetrics_corr$sample_id
```


```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cinmetrics_corr_2),
  #xlab = "Features", 
  #ylab = "Features"
)
```


```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
library(plotly)

fig <- plot_ly(
  cinmetrics_corr_2, x = ~HRDscore, y = ~break_points,
  # Hover text:
  text = ~rownames(cinmetrics_corr_2),
  color = ~HRDscore, size = ~break_points
) %>% layout(title = 'HRD34 vs Breakpoints')
fig
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig <- plot_ly(
  cinmetrics_corr_2, x = ~HRDscore, y = ~fga,
  # Hover text:
  text = ~rownames(cinmetrics_corr_2),
  color = ~HRDscore, size = ~break_points
) %>% layout(title = 'HRD34 vs fga')
fig
```


```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig <- plot_ly(
  cinmetrics_corr_2, x = ~HRDscore, y = ~cna,
  # Hover text:
  text = ~rownames(cinmetrics_corr_2),
  color = ~HRDscore, size = ~break_points
) %>% layout(title = 'HRD34 vs Cna')
fig
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig <- plot_ly(
  cinmetrics_corr_2, x = ~HRDscore, y = ~base_segments,
  # Hover text:
  text = ~rownames(cinmetrics_corr_2),
  color = ~HRDscore, size = ~break_points
) %>% layout(title = 'HRD34 vs Base_segments')
fig
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
library(stats)
library(tsne)
tsne = tsne(cinmetrics_corr_2, initial_dims = 2, k =2) 
tsne <- data.frame(tsne) 
```

#### CINmetrics tSNE plot

tSNE plot showing the clustering of normal and tumor tissue including HRD34 and the CINmetrics signatures.
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
pdb <- cbind(tsne,cinmetrics_corr$Sample.Type)
options(warn = -1)
fig <-  plot_ly(data = pdb ,x =  ~X1, y = ~X2, type = 'scatter', mode = 'markers', split = ~cinmetrics_corr$Sample.Type)

fig <- fig %>%
  layout(
    plot_bgcolor = "#e5ecf6"
  )

fig
```


#### CONCLUSION CINmetrics 

There is a correlation between HRD34 and the CINmetric signatures except for tai, which is stated by the cinmetrics publication that is a marker of normal or tumors in an earlier stage. In some types of cancer there seems to be a correlation between low HRD and lower breakpoints and cna, however the small data set limits this conclusions and more statistics are needed.

## CINdex

### CINdex at chromosome level
To mathematically and quantitatively describe these alternations we first locate
their genomic positions and measure their ranges. Such algorithms are referred to
as segmentation algorithms.

I run CINDex in the masked copy number segments from TCGA-SARC downloaded from GDC data commons. 
Masked copy number segments repport the segment mean which is the log2(copy-number/ 2). However, CINdex requires the CNV values. Therefore, in order to obtain the copy number value: 

$$
CNV = 2*2^{mean value}

$$
Settings:  CIN for a treshold gain of 2.25, threshold loss of 1.75, unnormalized (indicated by V.def=3), showing overall (gains and losses) CIN (indicated by V.mode=“sum’).

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Masked copy number data downloaded from TCGA has the mean normalized values, this is the log2(copy-number/ 2), meaning that the values equal to 0 are diploid, whilst the the ones with positive values are amplifies and below 0 are losses. 
maskCNV.SARC$norm <- 2*(2^maskCNV.SARC$Segment_Mean)
maskCNV.SARC$norm <- round(maskCNV.SARC$norm, 4)
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
#function to convert Copy number data into GRanges object
convert.to.gr.cnv <- function(cnv2) {
cnv2$Chromosome <- cnv2$Chromosome
# subset out SNPs with missing location
cnv2 <- cnv2[!(is.na(cnv2$Chromosome) | is.na(cnv2$Start)),]
# convert strand info to +,-,*

cnv2$strand <- c( "*")
#convert into GRanges object
cnv2.gr <- GRanges(cnv2$Chromosome, IRanges(cnv2$Start,cnv2$End),cnv2$strand, value = cnv2$norm)
return(cnv2.gr)
}
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
v1 <- list()
for(un1 in unique(maskCNV.SARC$Sample)){
  tmp <- maskCNV.SARC[maskCNV.SARC$Sample == un1, ]
  tmp <- tmp[!grepl("X", tmp$Chromosome),]
  tmp <- tmp[!grepl("Y", tmp$Chromosome),]
  cnv2gr <- convert.to.gr.cnv(cnv2 = tmp)
  
  #Subset only the probes that are in autosomes
  
  v1<- c(v1, cnv2gr)
}

cnvgr.tcga.sarc <- GRangesList(v1)
names(cnvgr.tcga.sarc)<- unique(maskCNV.SARC$Sample)
```

```{r cindex sum, include=FALSE, echo=FALSE, message=FALSE}
# Run CINdex by chromosome 
run.cin.chr(grl.seg = cnvgr.tcga.sarc, thr.gain=2.25, thr.loss=1.75, V.def=3, V.mode="sum")
```

```{r cindex amp, include=FALSE, echo=FALSE, message=FALSE}
# Run CINdex by chromosome 
run.cin.chr(grl.seg = cnvgr.tcga.sarc, thr.gain=2.25, thr.loss=1.75, V.def=3, V.mode="amp")
```

```{r cindex del , include=FALSE, echo=FALSE, message=FALSE}
# Run CINdex by chromosome 
run.cin.chr(grl.seg = cnvgr.tcga.sarc, thr.gain=2.25, thr.loss=1.75, V.def=3, V.mode="del")
```

```{r save cindex results, include=FALSE, echo=FALSE, message=FALSE}
#cin.sarc.results <- load("/Users/aliciapliego/Projects/HRDsarcoma/output_chr_cin/dataMatrix_2.25_1.75_unnormalized_sum.RData")
#cin.sarc.results.df <- dataMatrix
#cin.sarc.results.df <- t(cin.sarc.results.df)
#write.table(cin.sarc.results.df, "/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_sarc_results.tsv", sep="\t", quote = FALSE, row.names = T)

#cin.sarc.amp.results <- load("/Users/aliciapliego/Projects/HRDsarcoma/output_chr_cin/dataMatrix_2.25_1.75_unnormalized_amp.RData")
#cin.sarc.amp.results.df <- dataMatrix
#cin.sarc.amp.results.df <- t(cin.sarc.amp.results.df)
#write.table(cin.sarc.amp.results.df, "/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_sarc_amp_results.tsv", sep="\t", quote = FALSE, row.names = T)

cin.sarc.del.results <- load("/Users/aliciapliego/Projects/HRDsarcoma/output_chr_cin/dataMatrix_2.25_1.75_unnormalized_del.RData")
#cin.sarc.del.results.df <- dataMatrix
#cin.sarc.del.results.df <- t(cin.sarc.del.results.df)
#write.table(cin.sarc.del.results.df, "/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_sarc_del_results.tsv", sep="\t", quote = FALSE, row.names = T)
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
#Edit results in python and annotate them and add files back here
cindex_results_sum_sarc <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc.tsv', sep = '\t', header = TRUE)

rownames(cindex_results_sum_sarc) <- cindex_results_sum_sarc$Sample.ID
cindex_results_sum_sarc$Sample.ID <- NULL

cindex_results_sarc_meta <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_metadata.tsv', sep = '\t', header = TRUE)


```

```{r heatmap cindex all, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cindex_results_sum_sarc, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "CINdex TCGA",
  row_side_colors = cindex_results_sarc_meta[, c( "Sample.Type", "Tumor_type", "HRD34")]
                                             
  )
```

```{r , include=FALSE, echo=FALSE, message=FALSE}
cindex_results_sum_sarc_bp <- read.table('data/cindex_results_sarc_ptumor.tsv', sep = '\t', header = TRUE)
cindex_results_sum_sarc_meta_bp <- read.table('data/cindex_results_sarc_metadata_ptumor.tsv', sep = '\t', header = TRUE)
cindex_results_sum_sarc_bp$HRD34 <- cindex_results_sum_sarc_meta_bp$HRD34
cindex_results_sum_sarc_bp$Tumor_type <- cindex_results_sum_sarc_meta_bp$Tumor_type
```

### CINdex Primary tumor vs Normal 

```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
cindex_results_sum_sarc_normal <- cindex_results_sum_sarc
cindex_results_sum_sarc_normal$sums <- rowSums(cindex_results_sum_sarc_normal)
cindex_results_sum_sarc_normal$Sample.Type <- cindex_results_sarc_meta$Sample.Type
cindex_results_sum_sarc_normal$Tumor_type <- cindex_results_sarc_meta$Tumor_type

fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~sums, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1
```

#### Chromosome 1
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr1, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1
```

#### Chromosome 2
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr2, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1
```

#### Chromosome 3
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr3, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1
```

#### Chromosome 4
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr4, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1
```

#### Chromosome 5
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr5, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1
```

#### Chromosome 6
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr6, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1
```

#### Chromosome 7
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr7, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1
```

#### Chromosome 8
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr8, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1
```

#### Chromosome 9
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr9, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1
```

#### Chromosome 10
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr10, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1
```

#### Chromosome 11
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr11, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1


```

#### Chromosome 12
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr12, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1


```

#### Chromosome 13
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr13, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1


```

#### Chromosome 14
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr14, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1


```

#### Chromosome 15
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr15, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1

```

#### Chromosome 16
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr16, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1


```

#### Chromosome 17
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr17, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1


```

#### Chromosome 18
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr18, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1


```

#### Chromosome 19
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr19, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1


```

#### Chromosome 20
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr20, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1


```

#### Chromosome 21
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr21, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1


```

#### Chromosome 22
```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig1 <- plot_ly(cindex_results_sum_sarc_normal,x =~Tumor_type,  y = ~chr22, color = ~Sample.Type, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by Sample Type"), boxmode= 'group',
     yaxis = list())
fig1


```

#### Conclusions by chromosome

Summary t test for Primary tumor vs Normal tissue CIN per chromosome and sarcoma type:

  - chr1 Myxofibrosarcoma 1.0051178172904045e-05  
  - chr1 Dedifferentiated liposarcoma 2.3259123305144112e-05  
  - chr1 Leiomyosarcoma (LMS) 0.07726559073095456  
  - chr1 Undifferentiated Pleomorphic Sarcoma (UPS) 8.30682313755901e-05  
  - chr1 Leiomyosarcoma Uterus (LMS) 2.9454347187604525e-08  
  - chr1 Desmoid Tumor 0.05915240534187976  
  

  - chr2 Myxofibrosarcoma 1.1235849993147825e-06  
  - chr2 Dedifferentiated liposarcoma 0.0004337027393250114  
  - chr2 Leiomyosarcoma (LMS) 0.018345768886121636  
  - chr2 Undifferentiated Pleomorphic Sarcoma (UPS) 0.00012876188082388263  
  - chr2 Leiomyosarcoma Uterus (LMS) 2.994250397435886e-07  
  

  - chr3 Myxofibrosarcoma 0.003021518906776795  
  - chr3 Dedifferentiated liposarcoma 0.002529565715343376  
  - chr3 Leiomyosarcoma (LMS) 0.048443430264670115  
  - chr3 Undifferentiated Pleomorphic Sarcoma (UPS) 2.4888931880101017e-07  
  - chr3 Leiomyosarcoma Uterus (LMS) 0.0007832445935382686  
  
  
  - chr4 Myxofibrosarcoma 0.005768833126575008  
  - chr4 Dedifferentiated liposarcoma 0.012966546272164587   
  - chr4 Leiomyosarcoma (LMS) 0.0008891123729326209  
  - chr4 Undifferentiated Pleomorphic Sarcoma (UPS) 0.0011248851106732097  
  - chr4 Leiomyosarcoma Uterus (LMS) 0.0010550632066550466  
  
  
  - chr5 Myxofibrosarcoma 0.002382613677982469  
  - chr5 Dedifferentiated liposarcoma 0.00012092526692431861  
  - chr5 Leiomyosarcoma (LMS) 0.020178443700564243   
  - chr5 Undifferentiated Pleomorphic Sarcoma (UPS) 1.1719489355897795e-05  
  - chr5 Leiomyosarcoma Uterus (LMS) 0.002468702402472995  
  
  - chr6 Myxofibrosarcoma 0.0015183855856360307  
  - chr6 Dedifferentiated liposarcoma 0.00013367952442730791  
  - chr6 Leiomyosarcoma (LMS) 0.08235384864557929  
  - chr6 Undifferentiated Pleomorphic Sarcoma (UPS) 0.027859816057430965  
  - chr6 Leiomyosarcoma Uterus (LMS) 0.0006210236512757667  
 
 
  - chr7 Myxofibrosarcoma 0.007201984054418555  
  - chr7 Dedifferentiated liposarcoma 0.017919024287373676  
  - chr7 Undifferentiated Pleomorphic Sarcoma (UPS) 0.0010141101951651557  
  - chr7 Leiomyosarcoma Uterus (LMS) 0.00012098894031246307  
  - chr7 Malignant Peripheral Nerve Sheath Tumors (MPNST) 0.03454305221634289  
  
  - chr8 Myxofibrosarcoma 0.0007043080357777049  
  - chr8 Dedifferentiated liposarcoma 0.00025266111425827943  
  - chr8 Leiomyosarcoma (LMS) 0.01745740337443614  
  - chr8 Undifferentiated Pleomorphic Sarcoma (UPS) 2.9483136344668342e-05  
  - chr8 Leiomyosarcoma Uterus (LMS) 0.04178157427785321  
  
  
  - chr9 Myxofibrosarcoma 0.005236741034189228  
  - chr9 Dedifferentiated liposarcoma 0.0012711729733166728  
  - chr9 Leiomyosarcoma (LMS) 0.015614600591851319  
  - chr9 Undifferentiated Pleomorphic Sarcoma (UPS) 3.895613033074063e-05  
  - chr9 Leiomyosarcoma Uterus (LMS) 0.043728427307502905  
  - chr9 Synovial Sarcoma 0.06651531183539046  
 
 
  - chr10 Myxofibrosarcoma 6.449753729392839e-05  
  - chr10 Dedifferentiated liposarcoma 0.03125212093430745  
  - chr10 Leiomyosarcoma (LMS) 0.005665091238527053  
  - chr10 Undifferentiated Pleomorphic Sarcoma (UPS) 5.781787647203919e-06  
  - chr10 Leiomyosarcoma Uterus (LMS) 0.0005450671336302362  
  
  
  - chr11 Myxofibrosarcoma 6.7188181034577595e-06  
  - chr11 Dedifferentiated liposarcoma 0.002416002621385722  
  - chr11 Leiomyosarcoma (LMS) 0.09729621161145098  
  - chr11 Undifferentiated Pleomorphic Sarcoma (UPS) 0.0003494004093407973  
  - chr11 Leiomyosarcoma Uterus (LMS) 0.0010770184045157882   
  - chr11 Malignant Peripheral Nerve Sheath Tumors (MPNST) 0.06643067518115434   
  - chr11 Desmoid Tumor 0.06810324355436842   
  
  
  - chr12 Myxofibrosarcoma 0.014366284631583348  
  - chr12 Dedifferentiated liposarcoma 3.2415905252597296e-14  
  - chr12 Undifferentiated Pleomorphic Sarcoma (UPS) 0.02507697483990025  
  - chr12 Leiomyosarcoma Uterus (LMS) 0.008665117072330008  
  - chr12 Desmoid Tumor 0.008111260168514758  
  
  
  - chr13 Myxofibrosarcoma 0.003586918480256585  
  - chr13 Dedifferentiated liposarcoma 0.006419133870745385  
  - chr13 Leiomyosarcoma (LMS) 6.957046453632074e-07  
  - chr13 Undifferentiated Pleomorphic Sarcoma (UPS) 1.5539601750417358e-06  
  - chr13 Leiomyosarcoma Uterus (LMS) 5.7957081460068055e-05  
  
  - chr14 Myxofibrosarcoma 0.009761248358558482  
  - chr14 Dedifferentiated liposarcoma 0.026970286880360603  
  - chr14 Leiomyosarcoma (LMS) 0.012799892758824821  
  - chr14 Undifferentiated Pleomorphic Sarcoma (UPS) 0.0006192019746325063   
  - chr14 Leiomyosarcoma Uterus (LMS) 9.976968039632079e-05  
  - chr14 Malignant Peripheral Nerve Sheath Tumors (MPNST) 0.08401175797305843

  - chr15 Myxofibrosarcoma 0.01622931145767811  
  - chr15 Dedifferentiated liposarcoma 0.0004216506390224982  
  - chr15 Undifferentiated Pleomorphic Sarcoma (UPS) 0.037000919808370246  
  - chr15 Leiomyosarcoma Uterus (LMS) 0.0004612502665870319  
  - chr15 Synovial Sarcoma 0.07957780650461811  
  
  - chr16 Myxofibrosarcoma 1.0744045001346349e-06  
  - chr16 Dedifferentiated liposarcoma 0.014093798075528706  
  - chr16 Undifferentiated Pleomorphic Sarcoma (UPS) 1.2384297968903365e-07  
  - chr16 Leiomyosarcoma Uterus (LMS) 0.0003586027150035313  
  
  
  - chr17 Myxofibrosarcoma 0.00016706143146382046  
  - chr17 Dedifferentiated liposarcoma 0.014635244805819608  
  - chr17 Leiomyosarcoma (LMS) 5.513126538142558e-05  
  - chr17 Undifferentiated Pleomorphic Sarcoma (UPS) 4.314551172029698e-05  
  - chr17 Leiomyosarcoma Uterus (LMS) 5.806981190910449e-07  
  - chr17 Malignant Peripheral Nerve Sheath Tumors (MPNST) 0.0769663868720656   
  
  - chr18 Myxofibrosarcoma 0.00018699341569055174  
  - chr18 Dedifferentiated liposarcoma 0.004122484262375902  
  - chr18 Leiomyosarcoma (LMS) 0.06426213145455423   
  - chr18 Undifferentiated Pleomorphic Sarcoma (UPS) 0.0006745456508051654  
  - chr18 Leiomyosarcoma Uterus (LMS) 0.03540440974149572  
  
  - chr19 Myxofibrosarcoma 0.0013362641469685022   
  - chr19 Dedifferentiated liposarcoma 0.00022413612365476934  
  - chr19 Leiomyosarcoma (LMS) 0.0065150696672411626  
  - chr19 Undifferentiated Pleomorphic Sarcoma (UPS) 7.720739545963455e-06   
  - chr19 Leiomyosarcoma Uterus (LMS) 0.006527639097038919  
  - chr19 Malignant Peripheral Nerve Sheath Tumors (MPNST) 0.02649842174311059 
  
  - chr20 Myxofibrosarcoma 9.217286286265104e-06  
  - chr20 Dedifferentiated liposarcoma 0.003182307223562092  
  - chr20 Leiomyosarcoma (LMS) 0.0768587013758429  
  - chr20 Undifferentiated Pleomorphic Sarcoma (UPS) 1.6449280686267801e-06  
  - chr20 Leiomyosarcoma Uterus (LMS) 0.000792150198541933  
  
  - chr21 Myxofibrosarcoma 0.008470852609419549  
  - chr21 Dedifferentiated liposarcoma 0.06558708183585976  
  - chr21 Leiomyosarcoma (LMS) 0.09751000799818746  
  - chr21 Undifferentiated Pleomorphic Sarcoma (UPS) 0.0180264344716439  
  - chr21 Leiomyosarcoma Uterus (LMS) 0.021798537879322583
  
  - chr22 Myxofibrosarcoma 0.0004597661392020043  
  - chr22 Dedifferentiated liposarcoma 0.004719303723295408  
  - chr22 Leiomyosarcoma (LMS) 0.07807566846674234  
  - chr22 Undifferentiated Pleomorphic Sarcoma (UPS) 5.254216503150608e-07  
  - chr22 Leiomyosarcoma Uterus (LMS) 0.04898309593933952  
  - chr22 Malignant Peripheral Nerve Sheath Tumors (MPNST) 0.06305385142543547

In general in MFS we can see all the chromosomes with a higher CIN in the Primary tumor compared to Normal except in chromosme 15 and 12.

In ddLPS, chr 21,17,16,14, 10,7, and 4 do not show any difference between primary tumor and normal and chr 1,2,3,5,6,8,9,11,12,13,15,18,19,20 and 22 show a significant difference.

In LMS, chr 19,17,15,13, 12,10,7, and 4 show significant difference between primary tumor and normal, the rest not.

In UPS, chr 6,12,15 and 21 do not show significant difference between primary tumor and normal , the rest do.

In uLMS, chr 22, 21, 18, 9 and 8 do not show significant difference between primary tumor and normal , the rest do.

In MPNST there is no significance found due to the lack of cases.

### CIN only primary tumor

```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
cindex_results_sum_sarc_bp$sums <- rowSums(cindex_results_sum_sarc_bp[2:23])

fig1 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~sums, color = ~HRD34, type = "box",  notched=TRUE) %>% 
layout(boxmode = "group", title = list(text = "CINdex CIN score by HRD"), boxmode= 'group',
     yaxis = list())
fig1
```


```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig1 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr1, color = ~HRD34, type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig1
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}
fig2 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr2, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))

fig2
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}
fig3 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr3, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig3
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}
fig4 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr4, color = ~HRD34, type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig4
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}
fig5 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr5, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig5
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}
fig6 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr6, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig6
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}
fig7 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr7, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig7
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}
fig8 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr8, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig8

```


```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig9 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr9, color = ~HRD34, type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig9
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}
fig10 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr10, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig10
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}
fig11 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr11, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig11
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig12 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr12, color = ~HRD34, type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig12
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig13 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr13, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig13
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig14 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr14, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig14

```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig15 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr15, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig15
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig16 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr16, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))

fig16
```


```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig17 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr17, color = ~HRD34, type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig17
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig18 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr18, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig18
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig19 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr19, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig19
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig20 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr20, color = ~HRD34, type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig20
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig21 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr21, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig21
```
```{r , echo=FALSE, warning=FALSE, message=FALSE}

fig22 <- plot_ly(cindex_results_sum_sarc_bp,x =~Tumor_type,  y = ~chr22, color = ~HRD34,  type = "box",  notched=TRUE) %>%
layout(boxmode = "group",
     yaxis = list(range=c(0,150)))
fig22
```

#### Conclusions Statistics CIN in HRD high versus HRD low p val <0.01

  - chr1 Leiomyosarcoma (LMS) 0.0065428376313682425  
  
  - chr2 Myxofibrosarcoma 0.01083020827333234  
  - chr2 Leiomyosarcoma (LMS) 0.0049314298961821625  
  - chr2 Malignant Peripheral Nerve Sheath Tumors (MPNST) 0.002054414893073656  
  
  
  - chr3 Leiomyosarcoma (LMS) 0.0026379653450835272  
  - chr3 Undifferentiated Pleomorphic Sarcoma (UPS) 0.029284301193716326  
  - chr3 Leiomyosarcoma Uterus (LMS) 0.015558862651309383  
  
  
  - chr4 Leiomyosarcoma (LMS) 0.005923236803770181  
  - chr4 Leiomyosarcoma Uterus (LMS) 0.0038052383915415126  
  
  - chr5 Leiomyosarcoma (LMS) 0.0049756846202527235  
  - chr5 Undifferentiated Pleomorphic Sarcoma (UPS) 0.036197843924326144  
  - chr5 Malignant Peripheral Nerve Sheath Tumors (MPNST) 0.0038181446668229457 
  
  - chr8 Leiomyosarcoma (LMS) 0.00014155034152748492  
  - chr8 Malignant Peripheral Nerve Sheath Tumors (MPNST) 0.014965790107153838  
  
  - chr9 Leiomyosarcoma (LMS) 0.000900209759460778  
  
  - chr10 Leiomyosarcoma (LMS) 0.006192211687699704  
  - chr10 Malignant Peripheral Nerve Sheath Tumors (MPNST) 0.028731485887760224  
  
  - chr11 Myxofibrosarcoma 0.033469628089202716  
  - chr11 Leiomyosarcoma (LMS) 0.00025637022114651816  
  - chr11 Leiomyosarcoma Uterus (LMS) 0.008022853142300416  
  - chr11 Malignant Peripheral Nerve Sheath Tumors (MPNST) 0.0069482004422070105  
  
  - chr12 Malignant Peripheral Nerve Sheath Tumors (MPNST) 0.027766671064439048  
  
  - chr13 Dedifferentiated liposarcoma 0.006759242583420783  
  - chr13 Leiomyosarcoma (LMS) 0.00497129514585799  
  - chr13 Undifferentiated Pleomorphic Sarcoma (UPS) 0.049709329524452096  
  
  - chr14 Leiomyosarcoma (LMS) 2.232655775454952e-05  
  - chr14 Malignant Peripheral Nerve Sheath Tumors (MPNST) 0.00603510456533332  
  
  - chr15 Leiomyosarcoma (LMS) 0.0054219910857945565  
  
  - chr16 Myxofibrosarcoma 0.0339537664134872  
  
  - chr17 Leiomyosarcoma (LMS) 0.0036077874699638707  
  
  - chr18 Malignant Peripheral Nerve Sheath Tumors (MPNST) 0.028923506533018575  
  
  - chr19 Leiomyosarcoma (LMS) 0.018349316703571573  
  - chr19 Undifferentiated Pleomorphic Sarcoma (UPS) 0.04559800338381871  
  
  - chr19 Malignant Peripheral Nerve Sheath Tumors (MPNST) 0.045587208647691294  
  
  - chr20 Myxofibrosarcoma 0.027514053516250723  
  - chr20 Undifferentiated Pleomorphic Sarcoma (UPS) 0.038665137516787594
  - chr20 Malignant Peripheral Nerve Sheath Tumors (MPNST) 0.04584624868287656  
  
  - chr21 Leiomyosarcoma (LMS) 0.00023802599648183194  
  - chr22 Undifferentiated Pleomorphic Sarcoma (UPS) 0.040995095766073665
  
There are no differences in ddLPs, DT(no HRDhigh group), SS (no HRDhigh group). 


### Heatmaps CINdex  by sarcoma type 
#### MFS
```{r cindex mfs, include=FALSE, echo=FALSE, message=FALSE}
cindex_results_sarc_MFS <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_MFS.tsv', sep = '\t', header = TRUE)

rownames(cindex_results_sarc_MFS) <- cindex_results_sarc_MFS$Sample.ID
cindex_results_sarc_MFS$Sample.ID <- NULL

cindex_results_sarc_MFS_meta <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_metadata_MFS.tsv', sep = '\t', header = TRUE)
```

```{r heatmap cindex mfs, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cindex_results_sarc_MFS, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "CINdex TCGA MFS",
  row_side_colors = cindex_results_sarc_MFS_meta[, c("Tumor_type", "HRD34")]
                                             
  )
```

#### ddLPS
```{r cindex ddlps, include=FALSE, echo=FALSE, message=FALSE}
cindex_results_sarc_ddLPS <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_ddLPS.tsv', sep = '\t', header = TRUE)

rownames(cindex_results_sarc_ddLPS) <- cindex_results_sarc_ddLPS$Sample.ID
cindex_results_sarc_ddLPS$Sample.ID <- NULL

cindex_results_sarc_ddLPS_meta <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_metadata_ddLPS.tsv', sep = '\t', header = TRUE)
```

```{r heatmap cindex ddlps, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cindex_results_sarc_ddLPS, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "CINdex TCGA ddLPS",
  row_side_colors = cindex_results_sarc_ddLPS_meta[, c("Tumor_type", "HRD34")]
                                             
  )
```

#### LMS
```{r cindex lms, include=FALSE, echo=FALSE, message=FALSE}
cindex_results_sarc_LMS <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_LMS.tsv', sep = '\t', header = TRUE)

rownames(cindex_results_sarc_LMS) <- cindex_results_sarc_LMS$Sample.ID
cindex_results_sarc_LMS$Sample.ID <- NULL

cindex_results_sarc_LMS_meta <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_metadata_LMS.tsv', sep = '\t', header = TRUE)
```

```{r heatmap cindex lms, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cindex_results_sarc_LMS, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "CINdex TCGA LMS",
  row_side_colors = cindex_results_sarc_LMS_meta[, c("Tumor_type", "HRD34")]
                                             
  )
```

#### UPS
```{r cindex ups, include=FALSE, echo=FALSE, message=FALSE}
cindex_results_sarc_UPS <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_UPS.tsv', sep = '\t', header = TRUE)

rownames(cindex_results_sarc_UPS) <- cindex_results_sarc_UPS$Sample.ID
cindex_results_sarc_UPS$Sample.ID <- NULL

cindex_results_sarc_UPS_meta <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_metadata_UPS.tsv', sep = '\t', header = TRUE)
```

```{r heatmap cindex ups, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cindex_results_sarc_UPS, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "CINdex TCGA UPS",
  row_side_colors = cindex_results_sarc_UPS_meta[, c("Tumor_type", "HRD34")]
                                             
  )
```

#### uLMS
```{r cindex ulms, include=FALSE, echo=FALSE, message=FALSE}
cindex_results_sarc_uLMS <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_uLMS.tsv', sep = '\t', header = TRUE)

rownames(cindex_results_sarc_uLMS) <- cindex_results_sarc_uLMS$Sample.ID
cindex_results_sarc_uLMS$Sample.ID <- NULL

cindex_results_sarc_uLMS_meta <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_metadata_uLMS.tsv', sep = '\t', header = TRUE)
```

```{r heatmap cindex ulms, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cindex_results_sarc_uLMS, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "CINdex TCGA uLMS",
  row_side_colors = cindex_results_sarc_uLMS_meta[, c("Tumor_type", "HRD34")]
                                             
  )
```

#### SS
```{r cindex ss, include=FALSE, echo=FALSE, message=FALSE}
cindex_results_sarc_SS <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_SS.tsv', sep = '\t', header = TRUE)

rownames(cindex_results_sarc_SS) <- cindex_results_sarc_SS$Sample.ID
cindex_results_sarc_SS$Sample.ID <- NULL

cindex_results_sarc_SS_meta <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_metadata_SS.tsv', sep = '\t', header = TRUE)
```

```{r heatmap cindex ss, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cindex_results_sarc_SS, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "CINdex TCGA SS",
  row_side_colors = cindex_results_sarc_SS_meta[, c("Tumor_type", "HRD34")]
                                             
  )
```

#### MPNST
```{r cindex mpnst, include=FALSE, echo=FALSE, message=FALSE}
cindex_results_sarc_MPNST <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_MPNST.tsv', sep = '\t', header = TRUE)

rownames(cindex_results_sarc_MPNST) <- cindex_results_sarc_MPNST$Sample.ID
cindex_results_sarc_MPNST$Sample.ID <- NULL

cindex_results_sarc_MPNST_meta <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_metadata_MPNST.tsv', sep = '\t', header = TRUE)
```

```{r heatmap cindex mpnst, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cindex_results_sarc_MPNST, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "CINdex TCGA MPNST",
  row_side_colors = cindex_results_sarc_MPNST_meta[, c("Tumor_type", "HRD34")]
                                             
  )
```

#### DT
```{r cindex dt, include=FALSE, echo=FALSE, message=FALSE}
cindex_results_sarc_DT <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_DT.tsv', sep = '\t', header = TRUE)

rownames(cindex_results_sarc_DT) <- cindex_results_sarc_DT$Sample.ID
cindex_results_sarc_DT$Sample.ID <- NULL

cindex_results_sarc_DT_meta <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_metadata_DT.tsv', sep = '\t', header = TRUE)
```

```{r heatmap cindex dt, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply(cindex_results_sarc_DT, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "CINdex TCGA DT",
  row_side_colors = cindex_results_sarc_DT_meta[, c("Tumor_type", "HRD34")]
                                             
  )
```

### CINdex CINmetrics HRDness correlation

```{r cindex cinmetrics cindex correlation, include=FALSE, echo=FALSE, message=FALSE}
cindex_cinmetrix_corr<- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_cinmetrix_corr.tsv', sep = '\t', header = TRUE)
cindex_cinmetrix_corr<- cindex_cinmetrix_corr[!duplicated(cindex_cinmetrix_corr), ]
rownames(cindex_cinmetrix_corr) <- cindex_cinmetrix_corr$Sample.ID
cindex_cinmetrix_corr$Sample.ID <- NULL
cindex_cinmetrix_corr_log10 <- log10(cindex_cinmetrix_corr)
```

```{r heatmap cinmetrics cindex correlation , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
heatmaply_cor(
  cor(cindex_cinmetrix_corr),
  #xlab = "Features", 
  #ylab = "Features"
)
```

There is a High correlation between CINdex and the cna and break points, but there is also certain correlation with fga and base_segments.
#### All meassurements 
```{r all metrics, include=FALSE, echo=FALSE, message=FALSE}
#Filtered only by primary tumor 
all_metrics <- read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_cinmetrix_hrdscore_sarc.tsv', sep = '\t', header = TRUE)

```
##### fga in TCGA-SARC
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig <- plot_ly(all_metrics, y = ~fga, color = ~Tumor_type, type = "box",  text = all_metrics$Sample.ID, notched=TRUE) %>%
layout(
     xaxis = list(title = "",
     categoryorder = "array",
     categoryarray = c(
                         'UPS',
                         'MFS',
                         'uLMS',
                         'MPNST',
                         'LMS',
                         'ddLPS',
                         'SS',
                         'DT'))
)

fig
```

##### break_points in TCGA-SARC
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig <- plot_ly(all_metrics, y = ~break_points, color = ~Tumor_type, type = "box",  text = all_metrics$Sample.ID, notched=TRUE) %>%
layout(
     xaxis = list(title = "",
     categoryorder = "array",
     categoryarray = c(
                         'UPS',
                         'MFS',
                         'uLMS',
                         'MPNST',
                         'LMS',
                         'ddLPS',
                         'SS',
                         'DT'))
)

fig
```

##### cna in TCGA-SARC
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig <- plot_ly(all_metrics, y = ~cna, color = ~Tumor_type, type = "box",  text = all_metrics$Sample.ID, notched=TRUE) %>%
layout(
     xaxis = list(title = "",
     categoryorder = "array",
     categoryarray = c(
                         'UPS',
                         'MFS',
                         'uLMS',
                         'MPNST',
                         'LMS',
                         'ddLPS',
                         'SS',
                         'DT'))
)

fig
```

##### base_segments	 in TCGA-SARC
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig <- plot_ly(all_metrics, y = ~base_segments, color = ~Tumor_type, type = "box",  text = all_metrics$Sample.ID, notched=TRUE) %>%
layout(
     xaxis = list(title = "",
     categoryorder = "array",
     categoryarray = c(
                         'UPS',
                         'MFS',
                         'uLMS',
                         'MPNST',
                         'LMS',
                         'ddLPS',
                         'SS',
                         'DT'))
)

fig
```

##### CINdex	 in TCGA-SARC
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig <- plot_ly(all_metrics, y = ~CINdex_score, color = ~Tumor_type, type = "box",  text = all_metrics$Sample.ID, notched=TRUE) %>%
layout(
     xaxis = list(title = "",
     categoryorder = "array",
     categoryarray = c(
                         'UPS',
                         'MFS',
                         'uLMS',
                         'MPNST',
                         'LMS',
                         'ddLPS',
                         'SS',
                         'DT'))
)

fig
```

### CIN treatment 
```{r  include=FALSE, echo=FALSE, message=FALSE}
cin_treatment =  read.table('/Users/aliciapliego/Projects/HRDsarcoma/data/cindex_results_sarc_treatment_hrd.tsv', sep = '\t', header= TRUE)
```


```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig <- plot_ly(cin_treatment, x = ~treatment,y = ~total_cin, color = ~HRD34, type = "box", notched=TRUE) %>%
layout(boxmode = "group",
     xaxis = list(title = "CIN and treatment"
))

fig
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}
fig <- plot_ly(cin_treatment, x = ~treatment,y = ~total_cin, color = ~HRD34, type = "box", notched=TRUE) %>%
layout(boxmode = "group",
     xaxis = list(title = "CIN and treatment"
))

fig
```
