---
title: "RNAseq TCGA-SARC"
author: "Alicia Pliego"
output: 
  html_document:
    toc: true
    toc_float: true
    toc_collapsed: true
    toc_depth: 3
    number_sections: true
    theme: sandstone
---
<style>
body {
text-align: justify}
</style>

```{r setup, include=FALSE, echo=FALSE, message=FALSE}
library(DESeq2)
library(tximport)
library(pcaExplorer)
library(plotly)
library(pheatmap)
library(org.Hs.eg.db)
library(DOSE)
library(pathview)
library(clusterProfiler)
library(AnnotationHub)
library(ensembldb)
library(tidyverse)
library(EnhancedVolcano)
library(ComplexHeatmap)
library(DT)
```


# Differential expression analysis using DESeq2

#All sarcomas (ALL)
```{r,echo=FALSE, warning=FALSE, message=FALSE}
## Read in the metadata and the col data 
coldata <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/rnaseq/all_sarcoma_meta.txt', sep = '\t')
countdata <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/rnaseq/all_counts_filter.csv', sep = ',', check.names = FALSE)

rownames(countdata) <- countdata$ensembl
countdata$ensembl <- NULL

##Remove sum lines at the bottom of the coun dataframe
data_clean <- countdata[1:(nrow(countdata) - 5), ]

##Get symbol names for each enesembl gene ID
ens.str <- substr(rownames(data_clean), 1, 15)
data_clean$entrez <- mapIds(org.Hs.eg.db,
                            keys=ens.str,
                            column="SYMBOL",
                            keytype="ENSEMBL",
                            multiVals="first")

##Rearrange columns of the data-frame for the analysis
data_clean2<- data_clean
data_clean2<- na.omit(data_clean2)
data_clean3 <- data_clean2[!duplicated(data_clean2$entrez),]
rownames(data_clean3) <- data_clean3$entrez
data_clean3$entrez <- NULL

##Filtering counts with less than count reads
filter <- rowMeans(data_clean3) > 5
data_clean3Fil <- data_clean3[ filter, ]
```

### Samples to compare in the Differential Analysis
```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Plot distribution sarcomas HRD

```{r, echo=FALSE, warning=FALSE, message=FALSE}
##Run Deseq2 for HRDness
dds <- DESeqDataSetFromMatrix(countData = data_clean3Fil,
                              colData = coldata,
                              design = ~ HRDscore34)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#do filtering steps as for the clustering analysis
is_expressed <- assay(dds) >= 10
keep <- rowSums(assay(dds) >= 10) >= 2
dds <- dds[keep,]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
dds <- DESeq(dds)
res <- results(dds, contrast = c("HRDscore34", "HRD-high", "HRD-low"))
```

```{r,include=FALSE, echo=FALSE, warning=FALSE, message=FALSE}
# Get log2 counts
vsd <- vst(dds,blind=TRUE)
```


### Heatmap of the sample-to-sample distances 
I applied the dist function to the transpose of the transformed count matrix to get sample-to-sample distances.

```{r,  echo=FALSE, warning=FALSE, message=FALSE}
sampleDists <- dist(t(assay(vsd)))
```

```{r,  echo=FALSE, warning=FALSE, message=FALSE, fig.align='center'}
plotPCA(vsd,intgroup="HRDscore34")
```

```{r,  echo=FALSE, warning=FALSE, message=FALSE, fig.align='center'}
plotPCA(vsd,intgroup="histological_type")
```

There are two groups in the sarcoma PCA which do not cluster according to the sarcoma type, site of disease or HRD score.
```{r,  echo=FALSE, warning=FALSE, message=FALSE, fig.align='center'}
plotPCA(vsd,intgroup="site_of_disease")
```

## Comparison HRD high and HRD low  groups 
Summary of differentially expressed genes:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
resOrdered <- res[order(res$pvalue),]
summary(res)
```

Summary of differentially expressed genes.
Adjusting cut-off for and FDR of 0.05:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
res05 <- results(dds, alpha=0.05, contrast = c("HRDscore34", "HRD-high", "HRD-low"))
summary(res05)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
write.table(res05, "data/rnaseq/rnaseq_res05_all_sarcomas_AP.tsv",sep = "\t", quote = FALSE)
```

Cutoff values:
  pvalue  = 10e-16
  FCcutoff = 1.5

### Volcano plot 
Genes upregulated in the treated sample are in the right of the volcano plot, whilst downregulated genes are in the left hand side.
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}
EnhancedVolcano(res05,
    lab = rownames(res05),
    title = 'HRDhigh vs HRDlow',
    x = 'log2FoldChange',
    y = 'pvalue')

```

### Interactive volcano plot
```{r, echo=FALSE, warning=FALSE, message=FALSE}
res05_df <- as.data.frame(res05)
res05_df$log10P <- -log10(res05_df$pvalue) 
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}
library(plotly)

vline <- function(x = 0, color = "red") {
  list(
    type = "line", 
    y0 = 0, 
    y1 = 1, 
    yref = "paper",
    x0 = x, 
    x1 = x, 
    line = list(color = color)
  )
}
hline <- function(y = 0, color = "blue") {
  list(
    type = "line", 
    x0 = 0, 
    x1 = 1, 
    xref = "paper",
    y0 = y, 
    y1 = y, 
    line = list(color = color)
  )
}


fig <- plot_ly(data = res05_df, x = ~res05_df$log2FoldChange, y = ~res05_df$log10P,
               text= rownames(res05_df), color = ~res05_df$log2FoldChange) %>%
  layout(xaxis = list(title = 'log2FoldChange'), font=t, plot_bgcolor = "#e5ecf6",
         yaxis = list(title = '-log10(Pval)'), legend = list(title=list(text='Legend Title')), shapes = list(vline(2.5), hline(5)))

fig 
```

### Table of downregulated genes

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Further filter DE genes based on Padj = 0.01,  and FC>1.5(L2FC = 0.58)
res_filtered_pval <- subset(res05, padj <0.01)
res_filtered_all_up <- subset(res_filtered_pval, log2FoldChange >0.58)
res_filtered_all_up_df <- as.data.frame(res_filtered_all_up)

res_filtered_all_down <- subset(res_filtered_pval, log2FoldChange < -0.58)
res_filtered_all_down_df <- as.data.frame(res_filtered_all_down)
```
### Table of downregulated genes

```{r, echo=FALSE}
DT::datatable(res_filtered_all_down_df)
```

### Table of upregulated genes
```{r, echo=FALSE}
DT::datatable(res_filtered_all_up_df)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Rank genes and export for GSEA analysis
x <- res_filtered_pval
x$fcsign <- sign(x$log2FoldChange )
x$logP=-log10(x$pvalue)
x$metric= x$logP/x$fcsign
x$Gene <- rownames(x)
y<-x[,c("Gene", "metric")]
filtered <- na.omit(y)
#write.table(filtered ,file="/Users/aliciapliego/Projects/HRDsarcoma/data/rnaseq/DE_genes_all_sarcoma.rnk",quote=F,sep="\t",row.names=F)
```

## Heatmap of top 50 DE genes

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Select TOP 20 DE genes by pvalue
res_df<- as.data.frame(res_filtered_pval)
resOrdered   <- res_df[order(res_df$pvalue),]
resOrdered50 <- resOrdered[1:50,]
top50genes   <- rownames(resOrdered50)
top50genes   <- as.data.frame(top50genes)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Normalize results
dds <- estimateSizeFactors(dds)
normalized_counts <- counts(dds, normalized=T) %>% 
                     data.frame() %>%
                     rownames_to_column(var="gene") 
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
## This will bring in a column of gene symbols
top50_counts<- merge(normalized_counts, top50genes, by.x="gene", by.y="top50genes")
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

vst <- vst(dds, blind=FALSE)
vst <- assay(vst)
vst <- as.data.frame(vst)
vst_sig <- vst[rownames(vst) %in% top50_counts$gene,]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

heat <- scale(t(vst_sig))
```


```{r heatmap all sarcoma, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 8, fig.align='center'}

library(heatmaply)
coldata <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/rnaseq/all_sarcoma_meta.txt', sep = '\t')

heatmaply(heat, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = TRUE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "Heatmap of Differentially expressed genes in all sarcomas",
  row_side_colors = coldata[, c( "HRDscore34", "site_of_disease","annotation", "other_dx")]
                                             
  )
```

# Only taking the upper and lower quantiles:
## Plot histogram HRDscore 

```{r,echo=FALSE, warning=FALSE, message=FALSE}
metadata_sarcoma2 <- read.table("/Users/aliciapliego/Projects/HRDsarcoma/data/HRD_results_subgroupsSarcoma.tsv", sep = '\t', header = TRUE)

```

## Plot histogram HRDscore OV 

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}

general_mfs <- metadata_sarcoma2[(metadata_sarcoma2$Type) != "HGSOC",]
general_mfs <- general_mfs[(general_mfs$Type) != "TNBC",]
general_mfs <- general_mfs[(general_mfs$Type) != "CRC",]
general_mfs <- general_mfs[(general_mfs$Type) != "OS",]

fig <- plot_ly(x = ~general_mfs$HRD.sum, type = "histogram") %>% layout(bargap=0.1)

fig
```

```{r,echo=FALSE, warning=FALSE}
summary(general_mfs$HRD.sum)
```

Based on the quartiles, I will take two groups
  - Samples with a HRDsum <18 
  - Samples with a HRDsum >50
  
```{r,echo=FALSE, warning=FALSE, message=FALSE}
## Read in the metadata and the col data 
coldata <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/rnaseq/quartiles_all_sarcoma_metadata.txt', sep = '\t')
countdata <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/rnaseq/quartiles_all_sarcoma_counts.txt', sep = '\t', check.names = FALSE)

rownames(countdata) <- countdata$ensembl
countdata$ensembl <- NULL

##Remove sum lines at the bottom of the coun dataframe
data_clean <- countdata[1:(nrow(countdata) - 5), ]

##Get symbol names for each enesembl gene ID
ens.str <- substr(rownames(data_clean), 1, 15)
data_clean$entrez <- mapIds(org.Hs.eg.db,
                            keys=ens.str,
                            column="SYMBOL",
                            keytype="ENSEMBL",
                            multiVals="first")

##Rearrange columns of the data-frame for the analysis
data_clean2<- data_clean
data_clean2<- na.omit(data_clean2)
data_clean3 <- data_clean2[!duplicated(data_clean2$entrez),]
rownames(data_clean3) <- data_clean3$entrez
data_clean3$entrez <- NULL

##Filtering counts with less than count reads
filter <- rowMeans(data_clean3) > 5
data_clean3Fil <- data_clean3[ filter, ]
```

### Samples to compare in the Differential Analysis

```{r, echo=FALSE, warning=FALSE, message=FALSE}
##Run Deseq2 for HRDness
dds <- DESeqDataSetFromMatrix(countData = data_clean3Fil,
                              colData = coldata,
                              design = ~ HRD34)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#do filtering steps as for the clustering analysis
is_expressed <- assay(dds) >= 5
keep <- rowSums(assay(dds) >= 5) >= 2
dds <- dds[keep,]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
dds <- DESeq(dds)
res <- results(dds, contrast = c("HRD34", "HRD-high", "HRD-low"))
```

```{r,include=FALSE, echo=FALSE, warning=FALSE, message=FALSE}
# Get log2 counts
vsd <- vst(dds,blind=TRUE)
```


### Heatmap of the sample-to-sample distances 
I applied the dist function to the transpose of the transformed count matrix to get sample-to-sample distances.

```{r,  echo=FALSE, warning=FALSE, message=FALSE}
sampleDists <- dist(t(assay(vsd)))
```

```{r,  echo=FALSE, warning=FALSE, message=FALSE, fig.align='center'}
plotPCA(vsd,intgroup="HRD34")
```

```{r,  echo=FALSE, warning=FALSE, message=FALSE, fig.align='center'}
plotPCA(vsd,intgroup="Type")
```

```{r,  echo=FALSE, warning=FALSE, message=FALSE, fig.align='center'}
plotPCA(vsd,intgroup="site_of_disease")
```

```{r,  echo=FALSE, warning=FALSE, message=FALSE, fig.align='center'}
hclust_matrix <- data_clean3Fil %>% 
  # transpose the matrix so genes are as columns
  # apply scalling to each row of the matrix (samples)
  scale() %>% 
  # transpose back so genes are as rows again
  t()
gene_dist <- dist(hclust_matrix)
gene_hclust <- hclust(gene_dist, method = "complete")
```

```{r,  echo=FALSE, warning=FALSE, message=FALSE, fig.align='center'}
#gene_hclust <- hclust(gene_dist, method = "complete")
plot(gene_hclust, labels = FALSE)
abline(h = 188, col = "brown", lwd = 2) # add horizontal line to illustrate cutting dendrogram
cutree(gene_hclust, k = 2)

gene_cluster <- cutree(gene_hclust, k = 2) %>% 
  # turn the named vector into a tibble
  enframe() %>% 
  # rename some of the columns
  rename(gene = name, cluster = value)

write.table(gene_cluster ,"/Users/aliciapliego/Projects/HRDsarcoma/data/rnaseq/clusters_all_sarcoma.tsv",quote=F,sep="\t",row.names=T)

```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
##Run Deseq2 for HRDness
dds_clust <- DESeqDataSetFromMatrix(countData = data_clean3Fil,
                              colData = gene_cluster,
                              design = ~ cluster)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#do filtering steps as for the clustering analysis
is_expressed <- assay(dds_clust) >= 5
keep <- rowSums(assay(dds_clust) >= 5) >= 2
dds_clust <- dds_clust[keep,]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
dds_clust <- DESeq(dds_clust)
res <- results(dds_clust)
```

```{r,include=FALSE, echo=FALSE, warning=FALSE, message=FALSE}
# Get log2 counts
vsd <- vst(dds_clust,blind=TRUE)
```

```{r,  echo=FALSE, warning=FALSE, message=FALSE, fig.align='center'}
plotPCA(vsd,intgroup="cluster")
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Further filter DE genes based on Padj = 0.01,  and FC>1.5(L2FC = 0.58)
res_filtered_pval <- subset(res, padj <0.01)
res_filtered_all_up <- subset(res_filtered_pval, log2FoldChange >0.58)
res_filtered_all_up_df <- as.data.frame(res_filtered_all_up)

res_filtered_all_down <- subset(res_filtered_pval, log2FoldChange < -0.58)
res_filtered_all_down_df <- as.data.frame(res_filtered_all_down)
```

```{r}
write.table(res, "data/rnaseq/rnaseq_res_quartiles_clusters_all_sarcomas_AP.tsv",sep = "\t", quote = FALSE)
```

## GO Enrichment

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# remove duplicate IDS (here we use "SYMBOL", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

# Create a new dataframe df2 which has the respective entrez IDs for the gene symbols.
colnames(dedup_ids) = c("GeneSymbol", "EntrezID")
res$GeneSymbol <- rownames(res)
res2 <- as.data.frame(res)
df2 = merge(res2, dedup_ids, by = "GeneSymbol")

# Create a vector of the gene universe
kegg_gene_list = df2$log2FoldChange

# Name vector with ENTREZ ids
names(kegg_gene_list) = df2$EntrezID

# omit any NA values 
kegg_gene_list = na.omit(kegg_gene_list)

# sort the list in decreasing order (required for clusterProfiler)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_organism = "hsa"
kk2 = gseKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05,
               pAdjustMethod = "none",
               keyType       = "ncbi-geneid")

kk3 = gseMKEGG(geneList     = kegg_gene_list,
               organism     = kegg_organism,
               minGSSize    = 3,
               maxGSSize    = 800,
               pvalueCutoff = 0.05)
```

Plot results from KEGG pathways 
```{r, echo=FALSE}
out_kk3 = as.data.frame(kk2@result)
DT::datatable(out_kk3)
```

```{r , echo=FALSE, warning=FALSE, message=FALSE, fig.width= 10, fig.height = 10, fig.align='center'}

dotplot(kk2, showCategory = 15, title = "Enriched Pathways",font.size=14, split=".sign") + facet_grid(.~.sign)
```



### Table of downregulated genes

```{r, echo=FALSE}
DT::datatable(res_filtered_all_down_df)
```

### Table of upregulated genes
```{r, echo=FALSE}
DT::datatable(res_filtered_all_up_df)
```

## Comparison treated vs untreated groups 
Summary of differentially expressed genes:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
resOrdered <- res[order(res$pvalue),]
summary(res)
```

Summary of differentially expressed genes.
Adjusting cut-off for and FDR of 0.05:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
res05 <- results(dds, alpha=0.05, contrast = c("HRD34", "HRD-high", "HRD-low"))
summary(res05)
```

Cutoff values:
  pvalue  = 10e-16
  FCcutoff = 1.5

### Volcano plot 
Genes upregulated in the HRD high sample are in the right of the volcano plot, whilst downregulated genes are in the left hand side.
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}
EnhancedVolcano(res05,
    lab = rownames(res05),
    title = 'HRDhigh vs HRDlow',
    x = 'log2FoldChange',
    y = 'pvalue')

```

### Interactive volcano plot
```{r, echo=FALSE, warning=FALSE, message=FALSE}
res05_df <- as.data.frame(res05)
res05_df$log10P <- -log10(res05_df$pvalue) 
```



```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}
library(plotly)

vline <- function(x = 0, color = "red") {
  list(
    type = "line", 
    y0 = 0, 
    y1 = 1, 
    yref = "paper",
    x0 = x, 
    x1 = x, 
    line = list(color = color)
  )
}
hline <- function(y = 0, color = "blue") {
  list(
    type = "line", 
    x0 = 0, 
    x1 = 1, 
    xref = "paper",
    y0 = y, 
    y1 = y, 
    line = list(color = color)
  )
}


fig <- plot_ly(data = res05_df, x = ~res05_df$log2FoldChange, y = ~res05_df$log10P,
               text= rownames(res05_df), color = ~res05_df$log2FoldChange) %>%
  layout(xaxis = list(title = 'log2FoldChange'), font=t, plot_bgcolor = "#e5ecf6",
         yaxis = list(title = '-log10(Pval)'), legend = list(title=list(text='Legend Title')), shapes = list(vline(2.5), hline(5)))

fig 
```

We can observe MAGEC2, MAGEB1, MAGEA3 and PAGE2 upregulated in the HRD high group.
However, the number of upregulated genes is very low.
### Counts of most differentially expressed gene

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 5, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object
d <- plotCounts(dds, gene="WEE1", intgroup="HRDscore34", returnData=TRUE)

# Plotting the CTNNB1 normalized counts, using the samplenames (rownames of d as labels)
ggplot(d, aes(x = HRDscore34, y = count, color = HRDscore34)) + 
  geom_point(position=position_jitter(w = 0.1,h = 0)) +
  theme_bw() +
  ggtitle("WEE1") +
  theme(plot.title = element_text(hjust = 0.5))
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 5, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object
d <- plotCounts(dds, gene="MAGEB1", intgroup="HRDscore34", returnData=TRUE) 

# Plotting the CTNNB1 normalized counts, using the samplenames (rownames of d as labels)
ggplot(d, aes(x = HRDscore34, y = count, color = HRDscore34)) +
  geom_point(position=position_jitter(w = 0.1,h = 0)) +
  theme_bw() +
  ggtitle("MAGEB1") +
  theme(plot.title = element_text(hjust = 0.5))
```

We can observe that MAGEB1, which is one of the most upregulated genes is only up-regulated in three samples. This means that in MFS is difficult to make conclusions based on the RNA seq by comparing HRD high/low groups. 

### Counts of other Differenatially expressed genes

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object

par(mfrow=c(2,3))

plotCounts(dds, gene="WEE1", intgroup="HRDscore34")
plotCounts(dds, gene="PTEN", intgroup="HRDscore34")
plotCounts(dds, gene="TP53", intgroup="HRDscore34")
plotCounts(dds, gene="CDKN2A", intgroup="HRDscore34")
plotCounts(dds, gene="PARP1", intgroup="HRDscore34")
plotCounts(dds, gene="PAGE2", intgroup="HRDscore34")
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Further filter DE genes based on Padj = 0.01,  and FC>1.5(L2FC = 0.58)
res_filtered_pval <- subset(res05, padj <0.01)
res_filtered_all_up <- subset(res_filtered_pval, log2FoldChange >0.58)
res_filtered_all_up_df <- as.data.frame(res_filtered_all_up)

res_filtered_all_down <- subset(res_filtered_pval, log2FoldChange < -0.58)
res_filtered_all_down_df <- as.data.frame(res_filtered_all_down)
```

### Table of downregulated genes in Treated samples

```{r, echo=FALSE}
library(shiny)
library(DT)
```

```{r, echo=FALSE}
DT::datatable(res_filtered_all_down_df)
```

### Table of upregulated genes in Treated samples
```{r, echo=FALSE}
DT::datatable(res_filtered_all_up_df)
```


```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Rank genes and export for GSEA analysis
x <- res_filtered_pval
x$fcsign <- sign(x$log2FoldChange )
x$logP=-log10(x$pvalue)
x$metric= x$logP/x$fcsign
x$Gene <- rownames(x)
y<-x[,c("Gene", "metric")]
filtered <- na.omit(y)
#write.table(filtered ,file="DE_genes.rnk",quote=F,sep="\t",row.names=F)
```

## Heatmap of top 50 DE genes

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Select TOP 20 DE genes by pvalue
res_df<- as.data.frame(res_filtered_pval)
resOrdered   <- res_df[order(res_df$pvalue),]
resOrdered50 <- resOrdered[1:50,]
top50genes   <- rownames(resOrdered50)
top50genes   <- as.data.frame(top50genes)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Normalize results
dds <- estimateSizeFactors(dds)
normalized_counts <- counts(dds, normalized=T) %>% 
                     data.frame() %>%
                     rownames_to_column(var="gene") 
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
## This will bring in a column of gene symbols
top50_counts<- merge(normalized_counts, top50genes, by.x="gene", by.y="top50genes")
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

vst <- vst(dds, blind=FALSE)
vst <- assay(vst)
vst <- as.data.frame(vst)
vst_sig <- vst[rownames(vst) %in% top50_counts$gene,]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

heat <- t(scale(vst_sig))
```


```{r heatmap mfsspc, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}

library(heatmaply)
heatmaply(heat, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "Differentially expressed genes in TCGA MFS",
  row_side_colors = coldata[, c( "HRDscore34", "radio", "drug", "other_dx")]
  )
```



# Differential expression analysis using DESeq2
# Undeddifferentiated Pleomorphic Sarcoma (UPS)
```{r,echo=FALSE, warning=FALSE, message=FALSE}
## Read in the metadata and the col data 
coldata <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/rnaseq/UPS.meta.out.txt', sep = '\t')
countdata <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/rnaseq/UPS.counts.out.txt', sep = '\t', check.names = FALSE)

rownames(countdata) <- countdata$ensembl
countdata$ensembl <- NULL

##Remove sum lines at the bottom of the coun dataframe
data_clean <- countdata[1:(nrow(countdata) - 5), ]

##Get symbol names for each enesembl gene ID
ens.str <- substr(rownames(data_clean), 1, 15)
data_clean$entrez <- mapIds(org.Hs.eg.db,
                            keys=ens.str,
                            column="SYMBOL",
                            keytype="ENSEMBL",
                            multiVals="first")

##Rearrange columns of the data-frame for the analysis
data_clean2<- data_clean
data_clean2<- na.omit(data_clean2)
data_clean3 <- data_clean2[!duplicated(data_clean2$entrez),]
rownames(data_clean3) <- data_clean3$entrez
data_clean3$entrez <- NULL

##Filtering counts with less than count reads
filter <- rowMeans(data_clean3) > 5
data_clean3Fil <- data_clean3[ filter, ]
```
### Samples to compare in the Differential Analysis

```{r, echo=FALSE, warning=FALSE, message=FALSE}
##Run Deseq2 for HRDness
dds <- DESeqDataSetFromMatrix(countData = data_clean3Fil,
                              colData = coldata,
                              design = ~ HRDscore34)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#do filtering steps as for the clustering analysis
is_expressed <- assay(dds) >= 10
keep <- rowSums(assay(dds) >= 10) >= 2
dds <- dds[keep,]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
dds <- DESeq(dds)
res <- results(dds, contrast = c("HRDscore34", "HRDhigh", "HRDlow"))
```

## Comparison treated vs untreated groups 
Summary of differentially expressed genes:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
resOrdered <- res[order(res$pvalue),]
summary(res)
```

Summary of differentially expressed genes.
Adjusting cut-off for and FDR of 0.05:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
res05 <- results(dds, alpha=0.05, contrast = c("HRDscore34", "HRDhigh", "HRDlow"))
summary(res05)
```

Cutoff values:
  pvalue  = 10e-16
  FCcutoff = 1.5

### Volcano plot 
Genes upregulated in the HRD high sample are in the right of the volcano plot, whilst downregulated genes are in the left hand side.
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}
EnhancedVolcano(res05,
    lab = rownames(res05),
    title = 'HRDhigh vs HRDlow',
    x = 'log2FoldChange',
    y = 'pvalue')

```

### Interactive volcano plot
```{r, echo=FALSE, warning=FALSE, message=FALSE}
res05_df <- as.data.frame(res05)
res05_df$log10P <- -log10(res05_df$pvalue) 
```



```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}
library(plotly)

vline <- function(x = 0, color = "red") {
  list(
    type = "line", 
    y0 = 0, 
    y1 = 1, 
    yref = "paper",
    x0 = x, 
    x1 = x, 
    line = list(color = color)
  )
}
hline <- function(y = 0, color = "blue") {
  list(
    type = "line", 
    x0 = 0, 
    x1 = 1, 
    xref = "paper",
    y0 = y, 
    y1 = y, 
    line = list(color = color)
  )
}


fig <- plot_ly(data = res05_df, x = ~res05_df$log2FoldChange, y = ~res05_df$log10P,
               text= rownames(res05_df), color = ~res05_df$log2FoldChange) %>%
  layout(xaxis = list(title = 'log2FoldChange'), font=t, plot_bgcolor = "#e5ecf6",
         yaxis = list(title = '-log10(Pval)'), legend = list(title=list(text='Legend Title')), shapes = list(vline(2.5), hline(5)))

fig 
```

### Counts of most differentially expressed gene

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 5, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object
d <- plotCounts(dds, gene="WEE1", intgroup="HRDscore34", returnData=TRUE)

# Plotting the CTNNB1 normalized counts, using the samplenames (rownames of d as labels)
ggplot(d, aes(x = HRDscore34, y = count, color = HRDscore34)) + 
  geom_point(position=position_jitter(w = 0.1,h = 0)) +
  geom_text_repel(aes(label = rownames(d))) + 
  theme_bw() +
  ggtitle("WEE1") +
  theme(plot.title = element_text(hjust = 0.5))
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 5, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object
d <- plotCounts(dds, gene="MAGEA3", intgroup="HRDscore34", returnData=TRUE)

# Plotting the CTNNB1 normalized counts, using the samplenames (rownames of d as labels)
ggplot(d, aes(x = HRDscore34, y = count, color = HRDscore34)) + 
  geom_point(position=position_jitter(w = 0.1,h = 0)) +
  geom_text_repel(aes(label = rownames(d))) + 
  theme_bw() +
  ggtitle("MAGEA3") +
  theme(plot.title = element_text(hjust = 0.5))
```

### Counts of other Differenatially expressed genes
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object

par(mfrow=c(2,3))

plotCounts(dds, gene="WEE1", intgroup="HRDscore34")
plotCounts(dds, gene="PTEN", intgroup="HRDscore34")
plotCounts(dds, gene="TP53", intgroup="HRDscore34")
plotCounts(dds, gene="CDKN2A", intgroup="HRDscore34")
plotCounts(dds, gene="CDH12", intgroup="HRDscore34")
plotCounts(dds, gene="PAGE2", intgroup="HRDscore34")
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Further filter DE genes based on Padj = 0.01,  and FC>1.5(L2FC = 0.58)
res_filtered_pval <- subset(res05, padj <0.01)
res_filtered_all_up <- subset(res_filtered_pval, log2FoldChange >0.58)
res_filtered_all_up_df <- as.data.frame(res_filtered_all_up)

res_filtered_all_down <- subset(res_filtered_pval, log2FoldChange < -0.58)
res_filtered_all_down_df <- as.data.frame(res_filtered_all_down)
```

### Table of downregulated genes

```{r, echo=FALSE}
DT::datatable(res_filtered_all_down_df)
```

### Table of upregulated genes
```{r, echo=FALSE}
DT::datatable(res_filtered_all_up_df)
```


```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Rank genes and export for GSEA analysis
x <- res_filtered_pval
x$fcsign <- sign(x$log2FoldChange )
x$logP=-log10(x$pvalue)
x$metric= x$logP/x$fcsign
x$Gene <- rownames(x)
y<-x[,c("Gene", "metric")]
filtered <- na.omit(y)
#write.table(filtered ,file="DE_genes.rnk",quote=F,sep="\t",row.names=F)
```

## Heatmap of top 50 DE genes

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Select TOP 20 DE genes by pvalue
res_df<- as.data.frame(res_filtered_pval)
resOrdered   <- res_df[order(res_df$pvalue),]
resOrdered50 <- resOrdered[1:50,]
top50genes   <- rownames(resOrdered50)
top50genes   <- as.data.frame(top50genes)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Normalize results
dds <- estimateSizeFactors(dds)
normalized_counts <- counts(dds, normalized=T) %>% 
                     data.frame() %>%
                     rownames_to_column(var="gene") 
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
## This will bring in a column of gene symbols
top50_counts<- merge(normalized_counts, top50genes, by.x="gene", by.y="top50genes")
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

vst <- vst(dds, blind=FALSE)
vst <- assay(vst)
vst <- as.data.frame(vst)
vst_sig <- vst[rownames(vst) %in% top50_counts$gene,]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

heat <- t(scale(vst_sig))
```


```{r heatmap ups, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}

library(heatmaply)
heatmaply(heat, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "Differentially expressed genes in TCGA UPS",
  row_side_colors = coldata[, c( "HRDscore34", "other_dx")]
  )
```

# Differential expression analysis using DESeq2
# Leiomyosarcoma (LMS)
```{r,echo=FALSE, warning=FALSE, message=FALSE}
## Read in the metadata and the col data 
coldata <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/rnaseq/LMS.meta.out.txt', sep = '\t')
countdata <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/rnaseq/LMS.counts.out.txt', sep = '\t', check.names = FALSE)

rownames(countdata) <- countdata$ensembl
countdata$ensembl <- NULL

##Remove sum lines at the bottom of the coun dataframe
data_clean <- countdata[1:(nrow(countdata) - 5), ]

##Get symbol names for each enesembl gene ID
ens.str <- substr(rownames(data_clean), 1, 15)
data_clean$entrez <- mapIds(org.Hs.eg.db,
                            keys=ens.str,
                            column="SYMBOL",
                            keytype="ENSEMBL",
                            multiVals="first")

##Rearrange columns of the data-frame for the analysis
data_clean2<- data_clean
data_clean2<- na.omit(data_clean2)
data_clean3 <- data_clean2[!duplicated(data_clean2$entrez),]
rownames(data_clean3) <- data_clean3$entrez
data_clean3$entrez <- NULL

##Filtering counts with less than count reads
filter <- rowMeans(data_clean3) > 5
data_clean3Fil <- data_clean3[ filter, ]
```

### Samples to compare in the Differential Analysis

```{r, echo=FALSE, warning=FALSE, message=FALSE}
##Run Deseq2 for HRDness
dds <- DESeqDataSetFromMatrix(countData = data_clean3Fil,
                              colData = coldata,
                              design = ~ HRDscore34)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#do filtering steps as for the clustering analysis
is_expressed <- assay(dds) >= 10
keep <- rowSums(assay(dds) >= 10) >= 2
dds <- dds[keep,]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
dds <- DESeq(dds)
res <- results(dds, contrast = c("HRDscore34", "HRDhigh", "HRDlow"))
```

## Comparison treated vs untreated groups 
Summary of differentially expressed genes:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
resOrdered <- res[order(res$pvalue),]
summary(res)
```

Summary of differentially expressed genes.
Adjusting cut-off for and FDR of 0.05:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
res05 <- results(dds, alpha=0.05, contrast = c("HRDscore34", "HRDhigh", "HRDlow"))
summary(res05)
```

Cutoff values:
  pvalue  = 10e-16
  FCcutoff = 1.5

### Volcano plot 
Genes upregulated in the HRD high sample are in the right of the volcano plot, whilst downregulated genes are in the left hand side.
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}
EnhancedVolcano(res05,
    lab = rownames(res05),
    title = 'HRDhigh vs HRDlow',
    x = 'log2FoldChange',
    y = 'pvalue')

```

### Interactive volcano plot
```{r, echo=FALSE, warning=FALSE, message=FALSE}
res05_df <- as.data.frame(res05)
res05_df$log10P <- -log10(res05_df$pvalue) 
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}
library(plotly)

vline <- function(x = 0, color = "red") {
  list(
    type = "line", 
    y0 = 0, 
    y1 = 1, 
    yref = "paper",
    x0 = x, 
    x1 = x, 
    line = list(color = color)
  )
}
hline <- function(y = 0, color = "blue") {
  list(
    type = "line", 
    x0 = 0, 
    x1 = 1, 
    xref = "paper",
    y0 = y, 
    y1 = y, 
    line = list(color = color)
  )
}


fig <- plot_ly(data = res05_df, x = ~res05_df$log2FoldChange, y = ~res05_df$log10P,
               text= rownames(res05_df), color = ~res05_df$log2FoldChange) %>%
  layout(xaxis = list(title = 'log2FoldChange'), font=t, plot_bgcolor = "#e5ecf6",
         yaxis = list(title = '-log10(Pval)'), legend = list(title=list(text='Legend Title')), shapes = list(vline(2.5), hline(5)))

fig 
```

### Counts of most differentially expressed gene

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 5, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object
d <- plotCounts(dds, gene="WEE1", intgroup="HRDscore34", returnData=TRUE)

# Plotting the CTNNB1 normalized counts, using the samplenames (rownames of d as labels)
ggplot(d, aes(x = HRDscore34, y = count, color = HRDscore34)) + 
  geom_point(position=position_jitter(w = 0.1,h = 0)) +
  geom_text_repel(aes(label = rownames(d))) + 
  theme_bw() +
  ggtitle("WEE1") +
  theme(plot.title = element_text(hjust = 0.5))
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 5, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object
d <- plotCounts(dds, gene="MAGEA3", intgroup="HRDscore34", returnData=TRUE)

# Plotting the CTNNB1 normalized counts, using the samplenames (rownames of d as labels)
ggplot(d, aes(x = HRDscore34, y = count, color = HRDscore34)) + 
  geom_point(position=position_jitter(w = 0.1,h = 0)) +
  geom_text_repel(aes(label = rownames(d))) + 
  theme_bw() +
  ggtitle("MAGEA3") +
  theme(plot.title = element_text(hjust = 0.5))
```

### Counts of other Differenatially expressed genes
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 5, fig.align='center'}
# Save plotcounts to a data frame object

par(mfrow=c(2,3))

plotCounts(dds, gene="WEE1", intgroup="HRDscore34")
plotCounts(dds, gene="PTEN", intgroup="HRDscore34")
plotCounts(dds, gene="TP53", intgroup="HRDscore34")
plotCounts(dds, gene="CDKN2A", intgroup="HRDscore34")
plotCounts(dds, gene="CDH12", intgroup="HRDscore34")
plotCounts(dds, gene="PAGE2", intgroup="HRDscore34")
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Further filter DE genes based on Padj = 0.01,  and FC>1.5(L2FC = 0.58)
res_filtered_pval <- subset(res05, padj <0.01)
res_filtered_all_up <- subset(res_filtered_pval, log2FoldChange >0.58)
res_filtered_all_up_df <- as.data.frame(res_filtered_all_up)

res_filtered_all_down <- subset(res_filtered_pval, log2FoldChange < -0.58)
res_filtered_all_down_df <- as.data.frame(res_filtered_all_down)
```

### Table of downregulated genes

```{r, echo=FALSE}
DT::datatable(res_filtered_all_down_df)
```

### Table of upregulated genes
```{r, echo=FALSE}
DT::datatable(res_filtered_all_up_df)
```


```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Rank genes and export for GSEA analysis
x <- res_filtered_pval
x$fcsign <- sign(x$log2FoldChange )
x$logP=-log10(x$pvalue)
x$metric= x$logP/x$fcsign
x$Gene <- rownames(x)
y<-x[,c("Gene", "metric")]
filtered <- na.omit(y)
#write.table(filtered ,file="DE_genes.rnk",quote=F,sep="\t",row.names=F)
```
## Heatmap of top 50 DE genes

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Select TOP 20 DE genes by pvalue
res_df<- as.data.frame(res_filtered_pval)
resOrdered   <- res_df[order(res_df$pvalue),]
resOrdered50 <- resOrdered[1:50,]
top50genes   <- rownames(resOrdered50)
top50genes   <- as.data.frame(top50genes)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Normalize results
dds <- estimateSizeFactors(dds)
normalized_counts <- counts(dds, normalized=T) %>% 
                     data.frame() %>%
                     rownames_to_column(var="gene") 
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
## This will bring in a column of gene symbols
top50_counts<- merge(normalized_counts, top50genes, by.x="gene", by.y="top50genes")
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

vst <- vst(dds, blind=FALSE)
vst <- assay(vst)
vst <- as.data.frame(vst)
vst_sig <- vst[rownames(vst) %in% top50_counts$gene,]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

heat <- t(scale(vst_sig))
```


```{r heatmap lms, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}

library(heatmaply)
heatmaply(heat, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "Differentially expressed genes in TCGA LMS",
  row_side_colors = coldata[, c( "HRDscore34",  "other_dx")]
  )
```


# Differential expression analysis using DESeq2
# Uterine Leiomyosarcoma (uLMS)
```{r,echo=FALSE, warning=FALSE, message=FALSE}
## Read in the metadata and the col data 
coldata <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/rnaseq/uLMS.meta.out.txt', sep = '\t')
countdata <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/rnaseq/uLMS.counts.out.txt', sep = '\t', check.names = FALSE)

rownames(countdata) <- countdata$ensembl
countdata$ensembl <- NULL

##Remove sum lines at the bottom of the coun dataframe
data_clean <- countdata[1:(nrow(countdata) - 5), ]

##Get symbol names for each enesembl gene ID
ens.str <- substr(rownames(data_clean), 1, 15)
data_clean$entrez <- mapIds(org.Hs.eg.db,
                            keys=ens.str,
                            column="SYMBOL",
                            keytype="ENSEMBL",
                            multiVals="first")

##Rearrange columns of the data-frame for the analysis
data_clean2<- data_clean
data_clean2<- na.omit(data_clean2)
data_clean3 <- data_clean2[!duplicated(data_clean2$entrez),]
rownames(data_clean3) <- data_clean3$entrez
data_clean3$entrez <- NULL

##Filtering counts with less than count reads
filter <- rowMeans(data_clean3) > 5
data_clean3Fil <- data_clean3[ filter, ]
```

### Samples to compare in the Differential Analysis

```{r, echo=FALSE, warning=FALSE, message=FALSE}
##Run Deseq2 for HRDness
dds <- DESeqDataSetFromMatrix(countData = data_clean3Fil,
                              colData = coldata,
                              design = ~ HRDscore34)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#do filtering steps as for the clustering analysis
is_expressed <- assay(dds) >= 10
keep <- rowSums(assay(dds) >= 10) >= 2
dds <- dds[keep,]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
dds <- DESeq(dds)
res <- results(dds, contrast = c("HRDscore34", "HRDhigh", "HRDlow"))
```

## Comparison treated vs untreated groups 
Summary of differentially expressed genes:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
resOrdered <- res[order(res$pvalue),]
summary(res)
```

Summary of differentially expressed genes.
Adjusting cut-off for and FDR of 0.05:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
res05 <- results(dds, alpha=0.05, contrast = c("HRDscore34", "HRDhigh", "HRDlow"))
summary(res05)
```

Cutoff values:
  pvalue  = 10e-16
  FCcutoff = 1.5

### Volcano plot 
Genes upregulated in the HRD high sample are in the right of the volcano plot, whilst downregulated genes are in the left hand side.
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}
EnhancedVolcano(res05,
    lab = rownames(res05),
    title = 'HRDhigh vs HRDlow',
    x = 'log2FoldChange',
    y = 'pvalue')

```

### Interactive volcano plot
```{r, echo=FALSE, warning=FALSE, message=FALSE}
res05_df <- as.data.frame(res05)
res05_df$log10P <- -log10(res05_df$pvalue) 
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}
library(plotly)

vline <- function(x = 0, color = "red") {
  list(
    type = "line", 
    y0 = 0, 
    y1 = 1, 
    yref = "paper",
    x0 = x, 
    x1 = x, 
    line = list(color = color)
  )
}
hline <- function(y = 0, color = "blue") {
  list(
    type = "line", 
    x0 = 0, 
    x1 = 1, 
    xref = "paper",
    y0 = y, 
    y1 = y, 
    line = list(color = color)
  )
}


fig <- plot_ly(data = res05_df, x = ~res05_df$log2FoldChange, y = ~res05_df$log10P,
               text= rownames(res05_df), color = ~res05_df$log2FoldChange) %>%
  layout(xaxis = list(title = 'log2FoldChange'), font=t, plot_bgcolor = "#e5ecf6",
         yaxis = list(title = '-log10(Pval)'), legend = list(title=list(text='Legend Title')), shapes = list(vline(2.5), hline(5)))

fig 
```

### Counts of most differentially expressed gene



### Counts of other Differenatially expressed genes


```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Further filter DE genes based on Padj = 0.01,  and FC>1.5(L2FC = 0.58)
res_filtered_pval <- subset(res05, padj <0.01)
res_filtered_all_up <- subset(res_filtered_pval, log2FoldChange >0.58)
res_filtered_all_up_df <- as.data.frame(res_filtered_all_up)

res_filtered_all_down <- subset(res_filtered_pval, log2FoldChange < -0.58)
res_filtered_all_down_df <- as.data.frame(res_filtered_all_down)
```

### Table of downregulated genes


```{r, echo=FALSE}
DT::datatable(res_filtered_all_down_df)
```

### Table of upregulated genes
```{r, echo=FALSE}
DT::datatable(res_filtered_all_up_df)
```


```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Rank genes and export for GSEA analysis
x <- res_filtered_pval
x$fcsign <- sign(x$log2FoldChange )
x$logP=-log10(x$pvalue)
x$metric= x$logP/x$fcsign
x$Gene <- rownames(x)
y<-x[,c("Gene", "metric")]
filtered <- na.omit(y)
#write.table(filtered ,file="DE_genes.rnk",quote=F,sep="\t",row.names=F)
```

### Heatmap top 50 differentially expressed genes 

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Select TOP 20 DE genes by pvalue
res_df<- as.data.frame(res_filtered_pval)
resOrdered   <- res_df[order(res_df$pvalue),]
resOrdered50 <- resOrdered[1:50,]
top50genes   <- rownames(resOrdered50)
top50genes   <- as.data.frame(top50genes)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Normalize results
dds <- estimateSizeFactors(dds)
normalized_counts <- counts(dds, normalized=T) %>% 
                     data.frame() %>%
                     rownames_to_column(var="gene") 
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
## This will bring in a column of gene symbols
top50_counts<- merge(normalized_counts, top50genes, by.x="gene", by.y="top50genes")
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

vst <- vst(dds, blind=FALSE)
vst <- assay(vst)
vst <- as.data.frame(vst)
vst_sig <- vst[rownames(vst) %in% top50_counts$gene,]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

heat <- t(scale(vst_sig))
```


```{r heatmap ulms, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 10, fig.height = 7, fig.align='center'}

library(heatmaply)
heatmaply(heat, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "Differentially expressed genes in TCGA uLMS",
  row_side_colors = coldata[, c( "HRDscore34",  "other_dx")]
  )
```


# Differential expression analysis using DESeq2
# ddLPS
```{r,echo=FALSE, warning=FALSE, message=FALSE}
## Read in the metadata and the col data 
coldata <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/rnaseq/ddLPS.meta.out.txt', sep = '\t')
countdata <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/rnaseq/ddLPS.counts.out.txt', sep = '\t', check.names = FALSE)

rownames(countdata) <- countdata$ensembl
countdata$ensembl <- NULL

##Remove sum lines at the bottom of the coun dataframe
data_clean <- countdata[1:(nrow(countdata) - 5), ]

##Get symbol names for each enesembl gene ID
ens.str <- substr(rownames(data_clean), 1, 15)
data_clean$entrez <- mapIds(org.Hs.eg.db,
                            keys=ens.str,
                            column="SYMBOL",
                            keytype="ENSEMBL",
                            multiVals="first")

##Rearrange columns of the data-frame for the analysis
data_clean2<- data_clean
data_clean2<- na.omit(data_clean2)
data_clean3 <- data_clean2[!duplicated(data_clean2$entrez),]
rownames(data_clean3) <- data_clean3$entrez
data_clean3$entrez <- NULL

##Filtering counts with less than count reads
filter <- rowMeans(data_clean3) > 5
data_clean3Fil <- data_clean3[ filter, ]
```

### Samples to compare in the Differential Analysis

```{r, echo=FALSE, warning=FALSE, message=FALSE}
##Run Deseq2 for HRDness
dds <- DESeqDataSetFromMatrix(countData = data_clean3Fil,
                              colData = coldata,
                              design = ~ HRDscore34)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#do filtering steps as for the clustering analysis
is_expressed <- assay(dds) >= 10
keep <- rowSums(assay(dds) >= 10) >= 2
dds <- dds[keep,]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
dds <- DESeq(dds)
res <- results(dds, contrast = c("HRDscore34", "HRD-high", "HRD-low"))
```

## Comparison treated vs untreated groups 
Summary of differentially expressed genes:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
resOrdered <- res[order(res$pvalue),]
summary(res)
```

Summary of differentially expressed genes.
Adjusting cut-off for and FDR of 0.05:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
res05 <- results(dds, alpha=0.05, contrast = c("HRDscore34", "HRD-high", "HRD-low"))
summary(res05)
```

Cutoff values:
  pvalue  = 10e-16
  FCcutoff = 1.5

### Volcano plot 
Genes upregulated in the HRD high sample are in the right of the volcano plot, whilst downregulated genes are in the left hand side.
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}
EnhancedVolcano(res05,
    lab = rownames(res05),
    title = 'HRDhigh vs HRDlow',
    x = 'log2FoldChange',
    y = 'pvalue')

```

### Interactive volcano plot
```{r, echo=FALSE, warning=FALSE, message=FALSE}
res05_df <- as.data.frame(res05)
res05_df$log10P <- -log10(res05_df$pvalue) 
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}
library(plotly)

vline <- function(x = 0, color = "red") {
  list(
    type = "line", 
    y0 = 0, 
    y1 = 1, 
    yref = "paper",
    x0 = x, 
    x1 = x, 
    line = list(color = color)
  )
}
hline <- function(y = 0, color = "blue") {
  list(
    type = "line", 
    x0 = 0, 
    x1 = 1, 
    xref = "paper",
    y0 = y, 
    y1 = y, 
    line = list(color = color)
  )
}


fig <- plot_ly(data = res05_df, x = ~res05_df$log2FoldChange, y = ~res05_df$log10P,
               text= rownames(res05_df), color = ~res05_df$log2FoldChange) %>%
  layout(xaxis = list(title = 'log2FoldChange'), font=t, plot_bgcolor = "#e5ecf6",
         yaxis = list(title = '-log10(Pval)'), legend = list(title=list(text='Legend Title')), shapes = list(vline(2.5), hline(5)))

fig 
```

### Counts of most differentially expressed gene



### Counts of other Differenatially expressed genes
Other genes of interest such as  B-Catenin (CTNNB1) abd TP53 are also upregulated in the treated group.


```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Further filter DE genes based on Padj = 0.01,  and FC>1.5(L2FC = 0.58)
res_filtered_pval <- subset(res05, padj <0.01)
res_filtered_all_up <- subset(res_filtered_pval, log2FoldChange >0.58)
res_filtered_all_up_df <- as.data.frame(res_filtered_all_up)

res_filtered_all_down <- subset(res_filtered_pval, log2FoldChange < -0.58)
res_filtered_all_down_df <- as.data.frame(res_filtered_all_down)
```

### Table of downregulated genes


```{r, echo=FALSE}
DT::datatable(res_filtered_all_down_df)
```

### Table of upregulated genes
```{r, echo=FALSE}
DT::datatable(res_filtered_all_up_df)
```


```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Rank genes and export for GSEA analysis
x <- res_filtered_pval
x$fcsign <- sign(x$log2FoldChange )
x$logP=-log10(x$pvalue)
x$metric= x$logP/x$fcsign
x$Gene <- rownames(x)
y<-x[,c("Gene", "metric")]
filtered <- na.omit(y)
#write.table(filtered ,file="DE_genes.rnk",quote=F,sep="\t",row.names=F)
```

### Heatmap top 50 differentially expressed genes 

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Select TOP 20 DE genes by pvalue
res_df<- as.data.frame(res_filtered_pval)
resOrdered   <- res_df[order(res_df$pvalue),]
resOrdered50 <- resOrdered[1:50,]
top50genes   <- rownames(resOrdered50)
top50genes   <- as.data.frame(top50genes)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Normalize results
dds <- estimateSizeFactors(dds)
normalized_counts <- counts(dds, normalized=T) %>% 
                     data.frame() %>%
                     rownames_to_column(var="gene") 
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
## This will bring in a column of gene symbols
top50_counts<- merge(normalized_counts, top50genes, by.x="gene", by.y="top50genes")
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

vst <- vst(dds, blind=FALSE)
vst <- assay(vst)
vst <- as.data.frame(vst)
vst_sig <- vst[rownames(vst) %in% top50_counts$gene,]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

heat <- t(scale(vst_sig))
```


```{r heatmap ddlps, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}

library(heatmaply)
heatmaply(heat, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "Differentially expressed genes in TCGA ddLPS",
  row_side_colors = coldata[, c( "HRDscore34",  "other_dx")]
  )
```

# Differential expression analysis using DESeq2
# MPNST
```{r,echo=FALSE, warning=FALSE, message=FALSE}
## Read in the metadata and the col data 
coldata <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/rnaseq/MPNST.meta.out.txt', sep = '\t')
countdata <- read.csv('/Users/aliciapliego/Projects/HRDsarcoma/data/rnaseq/MPNST.counts.out.txt', sep = '\t', check.names = FALSE)

rownames(countdata) <- countdata$ensembl
countdata$ensembl <- NULL

##Remove sum lines at the bottom of the coun dataframe
data_clean <- countdata[1:(nrow(countdata) - 5), ]

##Get symbol names for each enesembl gene ID
ens.str <- substr(rownames(data_clean), 1, 15)
data_clean$entrez <- mapIds(org.Hs.eg.db,
                            keys=ens.str,
                            column="SYMBOL",
                            keytype="ENSEMBL",
                            multiVals="first")

##Rearrange columns of the data-frame for the analysis
data_clean2<- data_clean
data_clean2<- na.omit(data_clean2)
data_clean3 <- data_clean2[!duplicated(data_clean2$entrez),]
rownames(data_clean3) <- data_clean3$entrez
data_clean3$entrez <- NULL

##Filtering counts with less than count reads
filter <- rowMeans(data_clean3) > 5
data_clean3Fil <- data_clean3[ filter, ]
```

### Samples to compare in the Differential Analysis

```{r, echo=FALSE, warning=FALSE, message=FALSE}
##Run Deseq2 for HRDness
dds <- DESeqDataSetFromMatrix(countData = data_clean3Fil,
                              colData = coldata,
                              design = ~ HRDscore34)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#do filtering steps as for the clustering analysis
is_expressed <- assay(dds) >= 10
keep <- rowSums(assay(dds) >= 10) >= 2
dds <- dds[keep,]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
dds <- DESeq(dds)
res <- results(dds, contrast = c("HRDscore34", "HRD-high", "HRD-low"))
```

## Comparison treated vs untreated groups 
Summary of differentially expressed genes:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
resOrdered <- res[order(res$pvalue),]
summary(res)
```

Summary of differentially expressed genes.
Adjusting cut-off for and FDR of 0.05:
```{r, echo=FALSE, warning=FALSE, message=FALSE}
res05 <- results(dds, alpha=0.05, contrast = c("HRDscore34", "HRD-high", "HRD-low"))
summary(res05)
```

Cutoff values:
  pvalue  = 10e-16
  FCcutoff = 1.5

### Volcano plot 
Genes upregulated in the treated sample are in the right of the volcano plot, whilst downregulated genes are in the left hand side.
```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}
EnhancedVolcano(res05,
    lab = rownames(res05),
    title = 'HRDhigh vs HRDlow',
    x = 'log2FoldChange',
    y = 'pvalue')

```

### Interactive volcano plot
```{r, echo=FALSE, warning=FALSE, message=FALSE}
res05_df <- as.data.frame(res05)
res05_df$log10P <- -log10(res05_df$pvalue) 
```

```{r, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 8, fig.height = 9, fig.align='center'}
library(plotly)

vline <- function(x = 0, color = "red") {
  list(
    type = "line", 
    y0 = 0, 
    y1 = 1, 
    yref = "paper",
    x0 = x, 
    x1 = x, 
    line = list(color = color)
  )
}
hline <- function(y = 0, color = "blue") {
  list(
    type = "line", 
    x0 = 0, 
    x1 = 1, 
    xref = "paper",
    y0 = y, 
    y1 = y, 
    line = list(color = color)
  )
}


fig <- plot_ly(data = res05_df, x = ~res05_df$log2FoldChange, y = ~res05_df$log10P,
               text= rownames(res05_df), color = ~res05_df$log2FoldChange) %>%
  layout(xaxis = list(title = 'log2FoldChange'), font=t, plot_bgcolor = "#e5ecf6",
         yaxis = list(title = '-log10(Pval)'), legend = list(title=list(text='Legend Title')), shapes = list(vline(2.5), hline(5)))

fig 
```



```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Further filter DE genes based on Padj = 0.01,  and FC>1.5(L2FC = 0.58)
res_filtered_pval <- subset(res05, padj <0.01)
res_filtered_all_up <- subset(res_filtered_pval, log2FoldChange >0.58)
res_filtered_all_up_df <- as.data.frame(res_filtered_all_up)

res_filtered_all_down <- subset(res_filtered_pval, log2FoldChange < -0.58)
res_filtered_all_down_df <- as.data.frame(res_filtered_all_down)
```

### Table of downregulated genes


```{r, echo=FALSE}
DT::datatable(res_filtered_all_down_df)
```

### Table of upregulated genes
```{r, echo=FALSE}
DT::datatable(res_filtered_all_up_df)
```


```{r, echo=FALSE, warning=FALSE, message=FALSE}
# Rank genes and export for GSEA analysis
x <- res_filtered_pval
x$fcsign <- sign(x$log2FoldChange )
x$logP=-log10(x$pvalue)
x$metric= x$logP/x$fcsign
x$Gene <- rownames(x)
y<-x[,c("Gene", "metric")]
filtered <- na.omit(y)
#write.table(filtered ,file="DE_genes.rnk",quote=F,sep="\t",row.names=F)
```

### Heatmap top 50 differentially expressed genes 

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Select TOP 20 DE genes by pvalue
res_df<- as.data.frame(res_filtered_pval)
resOrdered   <- res_df[order(res_df$pvalue),]
resOrdered50 <- resOrdered[1:50,]
top50genes   <- rownames(resOrdered50)
top50genes   <- as.data.frame(top50genes)
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
#Normalize results
dds <- estimateSizeFactors(dds)
normalized_counts <- counts(dds, normalized=T) %>% 
                     data.frame() %>%
                     rownames_to_column(var="gene") 
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
## This will bring in a column of gene symbols
top50_counts<- merge(normalized_counts, top50genes, by.x="gene", by.y="top50genes")
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

vst <- vst(dds, blind=FALSE)
vst <- assay(vst)
vst <- as.data.frame(vst)
vst_sig <- vst[rownames(vst) %in% top50_counts$gene,]
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}

heat <- t(scale(vst_sig))
```


```{r heatmap mpnst, echo=FALSE, warning=FALSE, message=FALSE, fig.width= 7, fig.height = 7, fig.align='center'}

library(heatmaply)
heatmaply(heat, 
          
  scale_fill_gradient_fun = ggplot2::scale_fill_gradient2(
    low = "blue", 
    high = "red", 
    midpoint = 0.5, 
    limits = c(0, 1)
  ),
  Colv = FALSE,
  Rowv =TRUE,
  showticklabels = c(TRUE, FALSE),
  scale = "row",
  main = "Differentially expressed genes in TCGA MPNST",
  row_side_colors = coldata[, c( "HRDscore34",  "other_dx")]
  )
```